diff -uNr ecu_education_work_step1/swc-application/common/arxmls/RcCar.arxml ecu_education_work_step2/swc-application/common/arxmls/RcCar.arxml
--- ecu_education_work_step1/swc-application/common/arxmls/RcCar.arxml	2023-06-06 03:14:35.124601900 +0900
+++ ecu_education_work_step2/swc-application/common/arxmls/RcCar.arxml	2023-06-06 03:15:47.910091000 +0900
@@ -1462,6 +1462,9 @@
						</SWC-INTERNAL-BEHAVIOR>
					</INTERNAL-BEHAVIORS>
				</COMPLEX-DEVICE-DRIVER-SW-COMPONENT-TYPE>
+				<COMPLEX-DEVICE-DRIVER-SW-COMPONENT-TYPE>
+					<SHORT-NAME></SHORT-NAME>
+				</COMPLEX-DEVICE-DRIVER-SW-COMPONENT-TYPE>
				<APPLICATION-SW-COMPONENT-TYPE>
					<SHORT-NAME>AppLedControl</SHORT-NAME>
					<PORTS>
