diff -uNr ecu_education_work_step0/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml ecu_education_work_step1/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml
--- ecu_education_work_step0/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml	2023-06-06 03:13:42.197418600 +0900
+++ ecu_education_work_step1/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml	2023-06-06 03:14:34.725572100 +0900
@@ -180,10 +180,20 @@
       IB_CBL_TimingEvent_LedBlinker_500ms:
         DefinitionRef: RteEventToTaskMapping
         RteActivationOffset: 0.0
-        RtePositionInTask: 9
+        RtePositionInTask: 10
         RteEventRef: /RcCar/CddLedBlinker/IB_CBL/LedBlinker_timingEvent
         RteMappedToTaskRef: /Ecu_CBO/Os/ECU_CBO_TASK_100ms
         RteUsedOsAlarmRef: /Ecu_CBO/Os/ECU_CBO_ALARM_100ms
+    SWCI_AppLedControl:
+      DefinitionRef: RteSwComponentInstance
+      RteSoftwareComponentInstanceRef: /RcCar/Composition_RcCar/AppLedControl
+      IB_LC_TimingEvent_AppLedControl_500ms:
+        DefinitionRef: RteEventToTaskMapping
+        RteActivationOffset: 0.0
+        RtePositionInTask: 9
+        RteEventRef: /RcCar/AppLedControl/IB_LC/LedControl_timingEvent
+        RteMappedToTaskRef: /Ecu_CBO/Os/ECU_CBO_TASK_100ms
+        RteUsedOsAlarmRef: /Ecu_CBO/Os/ECU_CBO_ALARM_100ms
     RteOsInteraction:
       RteUsedOsActivation1:
         DefinitionRef: RteUsedOsActivation
diff -uNr ecu_education_work_step0/swc-application/1ecu/ecu_cbo/Makefile ecu_education_work_step1/swc-application/1ecu/ecu_cbo/Makefile
--- ecu_education_work_step0/swc-application/1ecu/ecu_cbo/Makefile	2023-06-06 03:13:42.200417400 +0900
+++ ecu_education_work_step1/swc-application/1ecu/ecu_cbo/Makefile	2023-06-06 03:14:34.729573400 +0900
@@ -205,6 +205,7 @@
 	CddBodyControl \
 	CddController \
 	CddRcCar \
+	AppLedControl \
 	OperationManager \
 	DriveManager \
 	Os_Hook \
diff -uNr ecu_education_work_step0/swc-application/common/arxmls/DataTypes.arxml ecu_education_work_step1/swc-application/common/arxmls/DataTypes.arxml
--- ecu_education_work_step0/swc-application/common/arxmls/DataTypes.arxml	2023-06-06 03:13:42.592310700 +0900
+++ ecu_education_work_step1/swc-application/common/arxmls/DataTypes.arxml	2023-06-06 03:14:35.121601400 +0900
@@ -181,6 +181,17 @@
 						</SW-DATA-DEF-PROPS-VARIANTS>
 					</SW-DATA-DEF-PROPS>
 				</IMPLEMENTATION-DATA-TYPE>
+				<IMPLEMENTATION-DATA-TYPE>
+					<SHORT-NAME>IDT_Led</SHORT-NAME>
+					<CATEGORY>VALUE</CATEGORY>
+					<SW-DATA-DEF-PROPS>
+						<SW-DATA-DEF-PROPS-VARIANTS>
+							<SW-DATA-DEF-PROPS-CONDITIONAL>
+								<BASE-TYPE-REF DEST="SW-BASE-TYPE">/BaseTypes/uint8</BASE-TYPE-REF>
+							</SW-DATA-DEF-PROPS-CONDITIONAL>
+						</SW-DATA-DEF-PROPS-VARIANTS>
+					</SW-DATA-DEF-PROPS>
+				</IMPLEMENTATION-DATA-TYPE>
 			</ELEMENTS>
 		</AR-PACKAGE>
 	</AR-PACKAGES>
diff -uNr ecu_education_work_step0/swc-application/common/arxmls/RcCar.arxml ecu_education_work_step1/swc-application/common/arxmls/RcCar.arxml
--- ecu_education_work_step0/swc-application/common/arxmls/RcCar.arxml	2023-06-06 03:13:42.596891800 +0900
+++ ecu_education_work_step1/swc-application/common/arxmls/RcCar.arxml	2023-06-06 03:14:35.124601900 +0900
@@ -89,6 +89,51 @@
 					</DATA-ELEMENTS>
 				</SENDER-RECEIVER-INTERFACE>
 				<SENDER-RECEIVER-INTERFACE>
+					<SHORT-NAME>IfSwState</SHORT-NAME>
+					<IS-SERVICE>false</IS-SERVICE>
+					<DATA-ELEMENTS>
+						<VARIABLE-DATA-PROTOTYPE>
+							<SHORT-NAME>dsw</SHORT-NAME>
+							<SW-DATA-DEF-PROPS>
+								<SW-DATA-DEF-PROPS-VARIANTS>
+									<SW-DATA-DEF-PROPS-CONDITIONAL>
+										<SW-IMPL-POLICY>STANDARD</SW-IMPL-POLICY>
+									</SW-DATA-DEF-PROPS-CONDITIONAL>
+								</SW-DATA-DEF-PROPS-VARIANTS>
+							</SW-DATA-DEF-PROPS>
+							<TYPE-TREF DEST="IMPLEMENTATION-DATA-TYPE">/ImplDataTypes/IDT_Switch</TYPE-TREF>
+						</VARIABLE-DATA-PROTOTYPE>
+						<VARIABLE-DATA-PROTOTYPE>
+							<SHORT-NAME>psw</SHORT-NAME>
+							<SW-DATA-DEF-PROPS>
+								<SW-DATA-DEF-PROPS-VARIANTS>
+									<SW-DATA-DEF-PROPS-CONDITIONAL>
+										<SW-IMPL-POLICY>STANDARD</SW-IMPL-POLICY>
+									</SW-DATA-DEF-PROPS-CONDITIONAL>
+								</SW-DATA-DEF-PROPS-VARIANTS>
+							</SW-DATA-DEF-PROPS>
+							<TYPE-TREF DEST="IMPLEMENTATION-DATA-TYPE">/ImplDataTypes/IDT_Switch</TYPE-TREF>
+						</VARIABLE-DATA-PROTOTYPE>
+					</DATA-ELEMENTS>
+				</SENDER-RECEIVER-INTERFACE>
+				<SENDER-RECEIVER-INTERFACE>
+					<SHORT-NAME>IfLedBlink</SHORT-NAME>
+					<IS-SERVICE>false</IS-SERVICE>
+					<DATA-ELEMENTS>
+						<VARIABLE-DATA-PROTOTYPE>
+							<SHORT-NAME>state</SHORT-NAME>
+							<SW-DATA-DEF-PROPS>
+								<SW-DATA-DEF-PROPS-VARIANTS>
+									<SW-DATA-DEF-PROPS-CONDITIONAL>
+										<SW-IMPL-POLICY>STANDARD</SW-IMPL-POLICY>
+									</SW-DATA-DEF-PROPS-CONDITIONAL>
+								</SW-DATA-DEF-PROPS-VARIANTS>
+							</SW-DATA-DEF-PROPS>
+							<TYPE-TREF DEST="IMPLEMENTATION-DATA-TYPE">/ImplDataTypes/IDT_Led</TYPE-TREF>
+						</VARIABLE-DATA-PROTOTYPE>
+					</DATA-ELEMENTS>
+				</SENDER-RECEIVER-INTERFACE>
+				<SENDER-RECEIVER-INTERFACE>
 					<SHORT-NAME>IfDriving</SHORT-NAME>
 					<IS-SERVICE>false</IS-SERVICE>
 					<DATA-ELEMENTS>
@@ -1360,6 +1405,57 @@
 				</COMPLEX-DEVICE-DRIVER-SW-COMPONENT-TYPE>
 				<COMPLEX-DEVICE-DRIVER-SW-COMPONENT-TYPE>
 					<SHORT-NAME>CddLedBlinker</SHORT-NAME>
+					<PORTS>
+						<R-PORT-PROTOTYPE>
+							<SHORT-NAME>LedBlinkIn</SHORT-NAME>
+							<REQUIRED-COM-SPECS>
+								<NONQUEUED-RECEIVER-COM-SPEC>
+									<DATA-ELEMENT-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfLedBlink/state</DATA-ELEMENT-REF>
+									<HANDLE-OUT-OF-RANGE>NONE</HANDLE-OUT-OF-RANGE>
+									<USES-END-TO-END-PROTECTION>false</USES-END-TO-END-PROTECTION>
+									<ALIVE-TIMEOUT>0.0</ALIVE-TIMEOUT>
+									<ENABLE-UPDATE>true</ENABLE-UPDATE>
+									<HANDLE-NEVER-RECEIVED>true</HANDLE-NEVER-RECEIVED>
+									<HANDLE-TIMEOUT-TYPE>NONE</HANDLE-TIMEOUT-TYPE>
+									<INIT-VALUE>
+										<NUMERICAL-VALUE-SPECIFICATION>
+											<SHORT-LABEL>IDT_Led</SHORT-LABEL>
+											<VALUE>0</VALUE>
+										</NUMERICAL-VALUE-SPECIFICATION>
+									</INIT-VALUE>
+								</NONQUEUED-RECEIVER-COM-SPEC>
+							</REQUIRED-COM-SPECS>
+							<REQUIRED-INTERFACE-TREF DEST="SENDER-RECEIVER-INTERFACE">/RcCar/IfLedBlink</REQUIRED-INTERFACE-TREF>
+						</R-PORT-PROTOTYPE>
+						<P-PORT-PROTOTYPE>
+							<SHORT-NAME>SwStateOut</SHORT-NAME>
+							<PROVIDED-COM-SPECS>
+								<NONQUEUED-SENDER-COM-SPEC>
+									<DATA-ELEMENT-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/dsw</DATA-ELEMENT-REF>
+									<HANDLE-OUT-OF-RANGE>NONE</HANDLE-OUT-OF-RANGE>
+									<USES-END-TO-END-PROTECTION>false</USES-END-TO-END-PROTECTION>
+									<INIT-VALUE>
+										<NUMERICAL-VALUE-SPECIFICATION>
+											<SHORT-LABEL>IDT_Switch</SHORT-LABEL>
+											<VALUE>0</VALUE>
+										</NUMERICAL-VALUE-SPECIFICATION>
+									</INIT-VALUE>
+								</NONQUEUED-SENDER-COM-SPEC>
+								<NONQUEUED-SENDER-COM-SPEC>
+									<DATA-ELEMENT-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/psw</DATA-ELEMENT-REF>
+									<HANDLE-OUT-OF-RANGE>NONE</HANDLE-OUT-OF-RANGE>
+									<USES-END-TO-END-PROTECTION>false</USES-END-TO-END-PROTECTION>
+									<INIT-VALUE>
+										<NUMERICAL-VALUE-SPECIFICATION>
+											<SHORT-LABEL>IDT_Switch</SHORT-LABEL>
+											<VALUE>0</VALUE>
+										</NUMERICAL-VALUE-SPECIFICATION>
+									</INIT-VALUE>
+								</NONQUEUED-SENDER-COM-SPEC>
+							</PROVIDED-COM-SPECS>
+							<PROVIDED-INTERFACE-TREF DEST="SENDER-RECEIVER-INTERFACE">/RcCar/IfSwState</PROVIDED-INTERFACE-TREF>
+						</P-PORT-PROTOTYPE>
+					</PORTS>
 					<INTERNAL-BEHAVIORS>
 						<SWC-INTERNAL-BEHAVIOR>
 							<SHORT-NAME>IB_CBL</SHORT-NAME>
@@ -1376,6 +1472,37 @@
 									<SHORT-NAME>LedBlinker</SHORT-NAME>
 									<MINIMUM-START-INTERVAL>0.0</MINIMUM-START-INTERVAL>
 									<CAN-BE-INVOKED-CONCURRENTLY>false</CAN-BE-INVOKED-CONCURRENTLY>
+									<DATA-RECEIVE-POINT-BY-ARGUMENTS>
+										<VARIABLE-ACCESS>
+											<SHORT-NAME>dataReceivePoint_CddLedBlinker_LedBlinkIn_state</SHORT-NAME>
+											<ACCESSED-VARIABLE>
+												<AUTOSAR-VARIABLE-IREF>
+													<PORT-PROTOTYPE-REF DEST="R-PORT-PROTOTYPE">/RcCar/CddLedBlinker/LedBlinkIn</PORT-PROTOTYPE-REF>
+													<TARGET-DATA-PROTOTYPE-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfLedBlink/state</TARGET-DATA-PROTOTYPE-REF>
+												</AUTOSAR-VARIABLE-IREF>
+											</ACCESSED-VARIABLE>
+										</VARIABLE-ACCESS>
+									</DATA-RECEIVE-POINT-BY-ARGUMENTS>
+									<DATA-SEND-POINTS>
+										<VARIABLE-ACCESS>
+											<SHORT-NAME>dataSendPoint_CddLedBlinker_SwStateOut_dsw</SHORT-NAME>
+											<ACCESSED-VARIABLE>
+												<AUTOSAR-VARIABLE-IREF>
+													<PORT-PROTOTYPE-REF DEST="P-PORT-PROTOTYPE">/RcCar/CddLedBlinker/SwStateOut</PORT-PROTOTYPE-REF>
+													<TARGET-DATA-PROTOTYPE-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/dsw</TARGET-DATA-PROTOTYPE-REF>
+												</AUTOSAR-VARIABLE-IREF>
+											</ACCESSED-VARIABLE>
+										</VARIABLE-ACCESS>
+										<VARIABLE-ACCESS>
+											<SHORT-NAME>dataSendPoint_CddLedBlinker_SwStateOut_psw</SHORT-NAME>
+											<ACCESSED-VARIABLE>
+												<AUTOSAR-VARIABLE-IREF>
+													<PORT-PROTOTYPE-REF DEST="P-PORT-PROTOTYPE">/RcCar/CddLedBlinker/SwStateOut</PORT-PROTOTYPE-REF>
+													<TARGET-DATA-PROTOTYPE-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/psw</TARGET-DATA-PROTOTYPE-REF>
+												</AUTOSAR-VARIABLE-IREF>
+											</ACCESSED-VARIABLE>
+										</VARIABLE-ACCESS>
+									</DATA-SEND-POINTS>
 									<SYMBOL>LedBlinker</SYMBOL>
 								</RUNNABLE-ENTITY>
 							</RUNNABLES>
@@ -1383,6 +1510,117 @@
 						</SWC-INTERNAL-BEHAVIOR>
 					</INTERNAL-BEHAVIORS>
 				</COMPLEX-DEVICE-DRIVER-SW-COMPONENT-TYPE>
+				<APPLICATION-SW-COMPONENT-TYPE>
+					<SHORT-NAME>AppLedControl</SHORT-NAME>
+					<PORTS>
+						<P-PORT-PROTOTYPE>
+							<SHORT-NAME>LedBlinkOut</SHORT-NAME>
+							<PROVIDED-COM-SPECS>
+								<NONQUEUED-SENDER-COM-SPEC>
+									<DATA-ELEMENT-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfLedBlink/state</DATA-ELEMENT-REF>
+									<HANDLE-OUT-OF-RANGE>NONE</HANDLE-OUT-OF-RANGE>
+									<USES-END-TO-END-PROTECTION>false</USES-END-TO-END-PROTECTION>
+									<INIT-VALUE>
+										<NUMERICAL-VALUE-SPECIFICATION>
+											<SHORT-LABEL>IDT_Led</SHORT-LABEL>
+											<VALUE>0</VALUE>
+										</NUMERICAL-VALUE-SPECIFICATION>
+									</INIT-VALUE>
+								</NONQUEUED-SENDER-COM-SPEC>
+							</PROVIDED-COM-SPECS>
+							<PROVIDED-INTERFACE-TREF DEST="SENDER-RECEIVER-INTERFACE">/RcCar/IfLedBlink</PROVIDED-INTERFACE-TREF>
+						</P-PORT-PROTOTYPE>
+						<R-PORT-PROTOTYPE>
+							<SHORT-NAME>SwStateIn</SHORT-NAME>
+							<REQUIRED-COM-SPECS>
+								<NONQUEUED-RECEIVER-COM-SPEC>
+									<DATA-ELEMENT-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/dsw</DATA-ELEMENT-REF>
+									<HANDLE-OUT-OF-RANGE>NONE</HANDLE-OUT-OF-RANGE>
+									<USES-END-TO-END-PROTECTION>false</USES-END-TO-END-PROTECTION>
+									<ALIVE-TIMEOUT>0.0</ALIVE-TIMEOUT>
+									<ENABLE-UPDATE>true</ENABLE-UPDATE>
+									<HANDLE-NEVER-RECEIVED>true</HANDLE-NEVER-RECEIVED>
+									<HANDLE-TIMEOUT-TYPE>NONE</HANDLE-TIMEOUT-TYPE>
+									<INIT-VALUE>
+										<NUMERICAL-VALUE-SPECIFICATION>
+											<SHORT-LABEL>IDT_Switch</SHORT-LABEL>
+											<VALUE>0</VALUE>
+										</NUMERICAL-VALUE-SPECIFICATION>
+									</INIT-VALUE>
+								</NONQUEUED-RECEIVER-COM-SPEC>
+								<NONQUEUED-RECEIVER-COM-SPEC>
+									<DATA-ELEMENT-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/psw</DATA-ELEMENT-REF>
+									<HANDLE-OUT-OF-RANGE>NONE</HANDLE-OUT-OF-RANGE>
+									<USES-END-TO-END-PROTECTION>false</USES-END-TO-END-PROTECTION>
+									<ALIVE-TIMEOUT>0.0</ALIVE-TIMEOUT>
+									<ENABLE-UPDATE>true</ENABLE-UPDATE>
+									<HANDLE-NEVER-RECEIVED>true</HANDLE-NEVER-RECEIVED>
+									<HANDLE-TIMEOUT-TYPE>NONE</HANDLE-TIMEOUT-TYPE>
+									<INIT-VALUE>
+										<NUMERICAL-VALUE-SPECIFICATION>
+											<SHORT-LABEL>IDT_Switch</SHORT-LABEL>
+											<VALUE>0</VALUE>
+										</NUMERICAL-VALUE-SPECIFICATION>
+									</INIT-VALUE>
+								</NONQUEUED-RECEIVER-COM-SPEC>
+							</REQUIRED-COM-SPECS>
+							<REQUIRED-INTERFACE-TREF DEST="SENDER-RECEIVER-INTERFACE">/RcCar/IfSwState</REQUIRED-INTERFACE-TREF>
+						</R-PORT-PROTOTYPE>
+					</PORTS>
+					<INTERNAL-BEHAVIORS>
+						<SWC-INTERNAL-BEHAVIOR>
+							<SHORT-NAME>IB_LC</SHORT-NAME>
+							<EVENTS>
+								<TIMING-EVENT>
+									<SHORT-NAME>LedControl_timingEvent</SHORT-NAME>
+									<START-ON-EVENT-REF DEST="RUNNABLE-ENTITY">/RcCar/AppLedControl/IB_LC/LedControl</START-ON-EVENT-REF>
+									<PERIOD>0.1</PERIOD>
+								</TIMING-EVENT>
+							</EVENTS>
+							<HANDLE-TERMINATION-AND-RESTART>NO-SUPPORT</HANDLE-TERMINATION-AND-RESTART>
+							<RUNNABLES>
+								<RUNNABLE-ENTITY>
+									<SHORT-NAME>LedControl</SHORT-NAME>
+									<MINIMUM-START-INTERVAL>0.0</MINIMUM-START-INTERVAL>
+									<CAN-BE-INVOKED-CONCURRENTLY>false</CAN-BE-INVOKED-CONCURRENTLY>
+									<DATA-RECEIVE-POINT-BY-ARGUMENTS>
+										<VARIABLE-ACCESS>
+											<SHORT-NAME>dataReceivePoint_LedControl_SwStateIn_dsw</SHORT-NAME>
+											<ACCESSED-VARIABLE>
+												<AUTOSAR-VARIABLE-IREF>
+													<PORT-PROTOTYPE-REF DEST="R-PORT-PROTOTYPE">/RcCar/AppLedControl/SwStateIn</PORT-PROTOTYPE-REF>
+													<TARGET-DATA-PROTOTYPE-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/dsw</TARGET-DATA-PROTOTYPE-REF>
+												</AUTOSAR-VARIABLE-IREF>
+											</ACCESSED-VARIABLE>
+										</VARIABLE-ACCESS>
+										<VARIABLE-ACCESS>
+											<SHORT-NAME>dataReceivePoint_LedControl_SwStateIn_psw</SHORT-NAME>
+											<ACCESSED-VARIABLE>
+												<AUTOSAR-VARIABLE-IREF>
+													<PORT-PROTOTYPE-REF DEST="R-PORT-PROTOTYPE">/RcCar/AppLedControl/SwStateIn</PORT-PROTOTYPE-REF>
+													<TARGET-DATA-PROTOTYPE-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfSwState/psw</TARGET-DATA-PROTOTYPE-REF>
+												</AUTOSAR-VARIABLE-IREF>
+											</ACCESSED-VARIABLE>
+										</VARIABLE-ACCESS>
+									</DATA-RECEIVE-POINT-BY-ARGUMENTS>
+									<DATA-SEND-POINTS>
+										<VARIABLE-ACCESS>
+											<SHORT-NAME>dataSendPoint_LedControl_LedBlinkOut_state</SHORT-NAME>
+											<ACCESSED-VARIABLE>
+												<AUTOSAR-VARIABLE-IREF>
+													<PORT-PROTOTYPE-REF DEST="P-PORT-PROTOTYPE">/RcCar/AppLedControl/LedBlinkOut</PORT-PROTOTYPE-REF>
+													<TARGET-DATA-PROTOTYPE-REF DEST="VARIABLE-DATA-PROTOTYPE">/RcCar/IfLedBlink/state</TARGET-DATA-PROTOTYPE-REF>
+												</AUTOSAR-VARIABLE-IREF>
+											</ACCESSED-VARIABLE>
+										</VARIABLE-ACCESS>
+									</DATA-SEND-POINTS>
+									<SYMBOL>LedControl</SYMBOL>
+								</RUNNABLE-ENTITY>
+							</RUNNABLES>
+							<SUPPORTS-MULTIPLE-INSTANTIATION>false</SUPPORTS-MULTIPLE-INSTANTIATION>
+						</SWC-INTERNAL-BEHAVIOR>
+					</INTERNAL-BEHAVIORS>
+				</APPLICATION-SW-COMPONENT-TYPE>
 				<COMPOSITION-SW-COMPONENT-TYPE>
 					<SHORT-NAME>Composition_RcCar</SHORT-NAME>
 					<COMPONENTS>
@@ -1410,6 +1648,10 @@
 							<SHORT-NAME>CddLedBlinker</SHORT-NAME>
 							<TYPE-TREF DEST="COMPLEX-DEVICE-DRIVER-SW-COMPONENT-TYPE">/RcCar/CddLedBlinker</TYPE-TREF>
 						</SW-COMPONENT-PROTOTYPE>
+						<SW-COMPONENT-PROTOTYPE>
+							<SHORT-NAME>AppLedControl</SHORT-NAME>
+							<TYPE-TREF DEST="APPLICATION-SW-COMPONENT-TYPE">/RcCar/AppLedControl</TYPE-TREF>
+						</SW-COMPONENT-PROTOTYPE>
 					</COMPONENTS>
 					<CONNECTORS>
 						<ASSEMBLY-SW-CONNECTOR>
@@ -1511,6 +1753,28 @@
 								<TARGET-R-PORT-REF DEST="R-PORT-PROTOTYPE">/RcCar/OperationManager/RCB3CommandClt</TARGET-R-PORT-REF>
 							</REQUESTER-IREF>
 						</ASSEMBLY-SW-CONNECTOR>
+						<ASSEMBLY-SW-CONNECTOR>
+							<SHORT-NAME>CddLedBlinker_SwStateOut_to_AppLedControl_SwStateIn</SHORT-NAME>
+							<PROVIDER-IREF>
+								<CONTEXT-COMPONENT-REF DEST="SW-COMPONENT-PROTOTYPE">/RcCar/Composition_RcCar/CddLedBlinker</CONTEXT-COMPONENT-REF>
+								<TARGET-P-PORT-REF DEST="P-PORT-PROTOTYPE">/RcCar/CddLedBlinker/SwStateOut</TARGET-P-PORT-REF>
+							</PROVIDER-IREF>
+							<REQUESTER-IREF>
+								<CONTEXT-COMPONENT-REF DEST="SW-COMPONENT-PROTOTYPE">/RcCar/Composition_RcCar/AppLedControl</CONTEXT-COMPONENT-REF>
+								<TARGET-R-PORT-REF DEST="R-PORT-PROTOTYPE">/RcCar/AppLedControl/SwStateIn</TARGET-R-PORT-REF>
+							</REQUESTER-IREF>
+						</ASSEMBLY-SW-CONNECTOR>
+						<ASSEMBLY-SW-CONNECTOR>
+							<SHORT-NAME>AppLedControl_LedBlinkOut_to_CddLedBlinker_LedBlinkIn</SHORT-NAME>
+							<PROVIDER-IREF>
+								<CONTEXT-COMPONENT-REF DEST="SW-COMPONENT-PROTOTYPE">/RcCar/Composition_RcCar/AppLedControl</CONTEXT-COMPONENT-REF>
+								<TARGET-P-PORT-REF DEST="P-PORT-PROTOTYPE">/RcCar/AppLedControl/LedBlinkOut</TARGET-P-PORT-REF>
+							</PROVIDER-IREF>
+							<REQUESTER-IREF>
+								<CONTEXT-COMPONENT-REF DEST="SW-COMPONENT-PROTOTYPE">/RcCar/Composition_RcCar/CddLedBlinker</CONTEXT-COMPONENT-REF>
+								<TARGET-R-PORT-REF DEST="R-PORT-PROTOTYPE">/RcCar/CddLedBlinker/LedBlinkIn</TARGET-R-PORT-REF>
+							</REQUESTER-IREF>
+						</ASSEMBLY-SW-CONNECTOR>
 					</CONNECTORS>
 				</COMPOSITION-SW-COMPONENT-TYPE>
 			</ELEMENTS>
diff -uNr ecu_education_work_step0/swc-application/common/swcs/AppLedControl.c ecu_education_work_step1/swc-application/common/swcs/AppLedControl.c
--- ecu_education_work_step0/swc-application/common/swcs/AppLedControl.c	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step1/swc-application/common/swcs/AppLedControl.c	2023-06-06 03:38:27.706427300 +0900
@@ -0,0 +1,96 @@
+/*
+ *  Copyright (C) 2023 by Daisuke Yasui
+ *
+ *  上記著作権者は，以下の(1)~(4)の条件を満たす場合に限り，本ソフトウェ
+ *  ア（本ソフトウェアを改変したものを含む．以下同じ）を使用・複製・改
+ *  変・再配布（以下，利用と呼ぶ）することを無償で許諾する．
+ *  (1) 本ソフトウェアをソースコードの形で利用する場合には，上記の著作
+ *      権表示，この利用条件および下記の無保証規定が，そのままの形でソー
+ *      スコード中に含まれていること．
+ *  (2) 本ソフトウェアを，ライブラリ形式など，他のソフトウェア開発に使
+ *      用できる形で再配布する場合には，再配布に伴うドキュメント（利用
+ *      者マニュアルなど）に，上記の著作権表示，この利用条件および下記
+ *      の無保証規定を掲載すること．
+ *  (3) 本ソフトウェアを，機器に組み込むなど，他のソフトウェア開発に使
+ *      用できない形で再配布する場合には，次のいずれかの条件を満たすこ
+ *      と．
+ *    (a) 再配布に伴うドキュメント（利用者マニュアルなど）に，上記の著
+ *        作権表示，この利用条件および下記の無保証規定を掲載すること．
+ *    (b) 再配布の形態を，別に定める方法によって，TOPPERSプロジェクトに
+ *        報告すること．
+ *  (4) 本ソフトウェアの利用により直接的または間接的に生じるいかなる損
+ *      害からも，上記著作権者およびTOPPERSプロジェクトを免責すること．
+ *      また，本ソフトウェアのユーザまたはエンドユーザからのいかなる理
+ *      由に基づく請求からも，上記著作権者およびTOPPERSプロジェクトを
+ *      免責すること．
+ *
+ *  本ソフトウェアは，AUTOSAR（AUTomotive Open System ARchitecture）仕
+ *  様に基づいている．上記の許諾は，AUTOSARの知的財産権を許諾するもので
+ *  はない．AUTOSARは，AUTOSAR仕様に基づいたソフトウェアを商用目的で利
+ *  用する者に対して，AUTOSARパートナーになることを求めている．
+ *
+ *  本ソフトウェアは，無保証で提供されているものである．上記著作権者お
+ *  よびTOPPERSプロジェクトは，本ソフトウェアに関して，特定の使用目的
+ *  に対する適合性も含めて，いかなる保証も行わない．また，本ソフトウェ
+ *  アの利用により直接的または間接的に生じたいかなる損害に関しても，そ
+ *  の責任を負わない．
+ *
+ */  
+
+/*
+ *  APP SW-C AppLedControl用の記述
+ */
+
+#include "Rte_AppLedControl.h"
+#include "ModelCarControl.h"
+#include "t_syslog.h"
+
+/*
+ *  APP SW-C AppLedControl用の初期化
+ */
+
+/*
+ *  Led制御ランナブル
+ */
+void
+LedControl(void)
+{
+	static uint8	active_led_val = LED1;
+	static boolean	active_led_up  = TRUE;
+
+	IDT_Switch dsw = 0;
+	IDT_Switch psw = 0;
+	Std_ReturnType ret = 0;
+
+	ret = Rte_Read_SwStateIn_dsw( &dsw );
+	if ( ret != 0 ) {
+		syslog_0( LOG_NOTICE, "Rte_Read_SwStateIn_dsw error" );
+	}
+
+	ret = Rte_Read_SwStateIn_psw( &psw );
+	if ( ret != 0 ) {
+		syslog_0( LOG_NOTICE, "Rte_Read_SwStateIn_psw error" );
+	}
+
+	syslog_2(LOG_NOTICE, "LedControl:dsw psw:%02x %02x", dsw, psw );
+	if ( dsw == 0 && psw == 0 ) {	/* dsw がすべて off かつ psw が off */
+		Rte_Write_LedBlinkOut_state( &active_led_val );
+	}
+	else {
+		Rte_Write_LedBlinkOut_state( 0 );
+	}
+
+	if(active_led_val == LED4){
+		active_led_up = FALSE;
+	}
+	if(active_led_val == LED1){
+		active_led_up = TRUE;
+	}
+
+	if(active_led_up == TRUE){
+		active_led_val = active_led_val << 1;
+	}
+	else {
+		active_led_val = active_led_val >> 1;
+	}
+}
diff -uNr ecu_education_work_step0/swc-application/common/swcs/CddLedBlinker.c ecu_education_work_step1/swc-application/common/swcs/CddLedBlinker.c
--- ecu_education_work_step0/swc-application/common/swcs/CddLedBlinker.c	2023-06-06 03:57:22.069416400 +0900
+++ ecu_education_work_step1/swc-application/common/swcs/CddLedBlinker.c	2023-06-06 03:59:10.285488700 +0900
@@ -65,11 +65,10 @@
 void
 LedBlinker(void)
 {
-	static uint8	active_led_val = LED1;
-	static boolean	active_led_up  = TRUE;
+	uint8	active_led_val = 0;
 
-	static IDT_Switch dsw_state = 0;
 	static IDT_Switch latch = 0; 
+	IDT_Switch dsw_state = 0;
 	IDT_Switch	psw_state = 0;
 	Std_ReturnType ret = 0;
 
@@ -84,33 +83,22 @@
 #else
 		dsw_state = (IDT_Switch)switch_dip_sense();
 #endif
-		syslog_1(LOG_NOTICE, "LedBlinker:dsw:%02x", dsw_state );
 		latch = 1;
 	}
 
 	psw_state = (IDT_Switch)switch_push_sense();
 
 	syslog_2(LOG_NOTICE, "LedBlinker:dsw psw:%02x %02x", dsw_state, psw_state );
-	if ( dsw_state == 0 && psw_state == 0 ) {	/* dsw がすべて off かつ psw が off */
-		led_out(active_led_val);
 
-		if(active_led_val == LED4){
-			active_led_up = FALSE;
-		}
-		if(active_led_val == LED1){
-			active_led_up = TRUE;
-		}
-
-		if(active_led_up == TRUE){
-			active_led_val = active_led_val << 1;
-		}
-		else {
-			active_led_val = active_led_val >> 1;
-		}
-	}
-	else {
-		led_out(0);
+	ret = Rte_Write_SwStateOut_dsw( &dsw_state );
+	ret += Rte_Write_SwStateOut_psw( &psw_state );
+	
+	ret = Rte_Read_LedBlinkIn_state( (IDT_Led*) &active_led_val );
+	if (ret != 0)
+	{
+		syslog_1(LOG_NOTICE, "LedBlinker:ret:%02x", ret );
 	}
+	led_out(active_led_val);
 #if SIMULATE_ON_ATHRILL
 	syslog_1(LOG_NOTICE, "LedBlinker:State:%02x", (*(uint8*)(0xffc10020)));
 #endif
