diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml ecu_education_work_step3/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml
--- ecu_education_work_step2/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml	2023-08-30 15:41:16.701824700 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_cbo/Ecuc_CBO.yaml	2023-09-11 05:46:34.987475600 +0900
@@ -46,12 +46,24 @@
       OsTaskSchedule: FULL
       OsTaskAutostart:
         OsTaskAppModeRef: /Ecu_CBO/Os/MainApp
+    :
+      DefinitionRef: OsTask
+      OsTaskActivation: 1
+      OsTaskPriority: 2
+      OsTaskStackSize: 1024
+      OsTaskSchedule: FULL
     RLIN3x_RX_ISR:
       DefinitionRef: OsIsr
       OsIsrInterruptPriority: 2
       OsIsrInterruptNumber: 158 #RLIN32_RX
       OsIsrCategory: CATEGORY_2
       OsIsrInterruptSource: ENABLE
+    :
+      DefinitionRef: OsIsr
+      OsIsrInterruptPriority: 1
+      OsIsrInterruptNumber: 25 #CAN_RX0
+      OsIsrCategory: CATEGORY_2
+      OsIsrInterruptSource: ENABLE
     ECU_CBO_ALARM_100ms:
       DefinitionRef: OsAlarm
       OsAlarmCounterRef: /Ecuc/Os/MAIN_HW_COUNTER
@@ -70,6 +82,12 @@
       OsAlarmAction:
         OsAlarmActivateTask:
           OsAlarmActivateTaskRef: /Ecu_CBO/Os/ECU_CBO_TASK_125us
+    :
+      DefinitionRef: OsAlarm
+      OsAlarmCounterRef: /Ecuc/Os/MAIN_HW_COUNTER
+      OsAlarmAction:
+        OsAlarmActivateTask:
+          OsAlarmActivateTaskRef: 
   Rte:
     RteGeneration:
       RteCodeVendorId: 65
@@ -220,6 +228,51 @@
         RteExpectedActivationOffset: 0.0
         RteExpectedTickDuration: 0.1
         RteActivationOsAlarmRef: /Ecu_CBO/Os/ECU_CBO_ALARM_100ms
+      :
+        DefinitionRef: RteUsedOsActivation
+        RteExpectedActivationOffset: 0.0
+        RteExpectedTickDuration: 0.02
+        RteActivationOsAlarmRef: 
+    :
+      DefinitionRef: RteBswModuleInstance
+      RteBswImplementationRef: /BSWScheduler/BswImplementations/bswImplementation0
+      RteBswExclusiveAreaImpl_exclusiveArea0:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/Reentrant_0
+      RteBswExclusiveAreaImpl_exclusiveArea1:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/Reentrant_1
+      RteBswExclusiveAreaImpl_exclusiveArea2:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/Reentrant_2
+      :
+        DefinitionRef: RteBswEventToTaskMapping
+        RteBswPositionInTask: 0
+        RteBswActivationOffset: 0.02
+        RteBswEventRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/TimingEvent_MainFunctionRx
+        RteBswMappedToTaskRef: 
+        RteBswUsedOsAlarmRef: 
+    :
+      DefinitionRef: RteBswModuleInstance
+      RteBswImplementationRef: /BSWScheduler/BswImplementations/bswImplementation1
+      RteBswExclusiveAreaImpl_exclusiveArea3:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Can/InternalBehavior_0/Reentrant
+    :
+      DefinitionRef: RteBswModuleInstance
+      RteBswImplementationRef: /BSWScheduler/BswImplementations/bswImplementation2
+      RteBswExclusiveAreaImpl_exclusiveArea4:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/CanIf/InternalBehavior_0/Reentrant_0
+      RteBswExclusiveAreaImpl_exclusiveArea5:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/CanIf/InternalBehavior_0/Reentrant_1
     RteBswGeneral:
       RteSchMVersionInfoApi: true
       RteUseComShadowSignalApi: false
@@ -230,3 +283,174 @@
       Pdu0:
         DefinitionRef: Pdu
         PduLength: 8
+  EcuM:
+    EcuMConfiguration0:
+      DefinitionRef: EcuMConfiguration
+      EcuMCommonConfiguration0:
+        DefinitionRef: EcuMCommonConfiguration
+        EcuMWakeupSource0:
+          DefinitionRef: EcuMWakeupSource
+          EcuMWakeupSourceId: 0
+          EcuMWakeupSourcePolling: false
+  Com:
+    ComConfig:
+      ComConfigurationId: 0
+      :
+        DefinitionRef: ComSignal
+        ComHandleId: 0
+        ComBitPosition: 0
+        ComBitSize: 8
+        ComSignalType: UINT8
+        ComTransferProperty: TRIGGERED_ON_CHANGE
+        ComSignalEndianness: LITTLE_ENDIAN
+        ComSystemTemplateSystemSignalRef: 
+      :
+        DefinitionRef: ComSignal
+        ComHandleId: 1
+        ComBitPosition: 8
+        ComBitSize: 8
+        ComSignalType: UINT8
+        ComSignalEndianness: LITTLE_ENDIAN
+        ComTransferProperty: TRIGGERED_ON_CHANGE
+        ComSystemTemplateSystemSignalRef: 
+      ComIPdu0:
+        DefinitionRef: ComIPdu
+        ComIPduHandleId: 0
+        ComIPduDirection: RECEIVE
+        ComIPduSignalProcessing: IMMEDIATE
+        ComIPduType: NORMAL
+        ComIPduGroupRef: /Ecu_CBO/Com/ComConfig/ComIPduGroup0
+        ComIPduSignalRef:
+          - /Ecu_CBO/Com/ComConfig/ComSignal0
+          - /Ecu_CBO/Com/ComConfig/ComSignal1
+        ComPduIdRef: /Ecu_CBO/EcuC/EcucPduCollection/Pdu0
+      ComIPduGroup0:
+        DefinitionRef: ComIPduGroup
+        ComIPduGroupHandleId: 0
+      ComTimeBase:
+        ComGwTimeBase: 0.02
+        ComRxTimeBase: 0.02
+        ComTxTimeBase: 0.02
+    ComGeneral:
+      ComSupportedIPduGroups: 1
+      ComVersionInfoApi: false
+      ComConfigurationUseDet: true
+  PduR:
+    PduRGeneral:
+      PduRZeroCostOperation: true
+      PduRDevErrorDetect: false
+      PduRVersionInfoApi: false
+    PduRRoutingTables:
+      PduRConfigurationId: 0
+      PduRRoutingTable:
+        PduRIsMinimumRouting: false
+        PduRRoutingPath_0:
+          DefinitionRef: PduRRoutingPath
+          PduRSrcPdu0:
+            DefinitionRef: PduRSrcPdu
+            PduRSourcePduHandleId: 0
+            PduRSrcPduRef: /Ecu_CBO/EcuC/EcucPduCollection/Pdu0
+          PduRDestPdu10:
+            DefinitionRef: PduRDestPdu
+            PduRDestPduHandleId: 0
+            PduRDestPduRef: /Ecu_CBO/EcuC/EcucPduCollection/Pdu0
+  CanIf:
+    CanIfPublicCfg:
+      CanIfPublicNumberOfCanHwUnits: 1
+      CanIfPublicCancelTransmitSupport: true
+      CanIfPublicChangeBaudrateSupport: true
+      CanIfPublicDevErrorDetect: true
+      CanIfPublicHandleTypeEnum: UINT8
+      CanIfPublicMultipleDrvSupport: false
+      CanIfPublicPnSupport: false
+      CanIfPublicReadRxPduDataApi: true
+      CanIfPublicReadRxPduNotifyStatusApi: true
+      CanIfPublicReadTxPduNotifyStatusApi: true
+      CanIfPublicSetDynamicTxIdApi: false
+      CanIfPublicTxBuffering: false
+      CanIfPublicTxConfirmPollingSupport: true
+      CanIfPublicVersionInfoApi: false
+      CanIfPublicWakeupCheckValidSupport: true
+    CanIfPrivateCfg:
+      CanIfPrivateDlcCheck: true
+      CanIfSupportTTCAN: false
+    CanIfDispatchCfg:
+      CanIfDispatchUserCtrlBusOffName: CanSM_ControllerBusOff
+      CanIfDispatchUserCtrlBusOffUL: CAN_SM
+      CanIfDispatchUserCtrlModeIndicationName: CanSM_ControllerModeIndication
+      CanIfDispatchUserCtrlModeIndicationUL: CAN_SM
+      CanIfDispatchUserValidateWakeupEventName: EcuM_ValidateWakeupEvent
+      CanIfDispatchUserValidateWakeupEventUL: ECUM
+    CanIfCtrlDrvCfg:
+      CanIfCtrlDrvTxCancellation: false
+      CanIfCtrlDrvInitHohConfigRef: /Ecu_CBO/CanIf/CanIfInitCfg/CanIfInitHohCfg
+      CanIfCtrlDrvNameRef: /Ecu_CBO/Can/CanGeneral
+      CanIfCtrlCfg0:
+        DefinitionRef: CanIfCtrlCfg
+        CanIfCtrlId: 0
+        CanIfCtrlWakeupSupport: true
+        CanIfCtrlCanCtrlRef: /Ecu_CBO/Can/CanConfigSet0/CanController0
+    CanIfInitCfg:
+      CanIfInitCfgSet: main_canif_config
+      CanIfInitHohCfg:
+        CanIfInitRefCfgSet: /Ecu_CBO/Can/CanConfigSet0
+        CanIfHrhCfg0:
+          DefinitionRef: CanIfHrhCfg
+          CanIfHrhSoftwareFilter: false
+          CanIfHrhCanCtrlIdRef: /Ecu_CBO/CanIf/CanIfCtrlDrvCfg/CanIfCtrlCfg0
+          CanIfHrhIdSymRef: /Ecu_CBO/Can/CanConfigSet0/CanHardwareObject0
+      :
+        DefinitionRef: CanIfRxPduCfg
+        CanIfRxPduCanId: 1
+        CanIfRxPduDlc: 8
+        CanIfRxPduId: 0
+        CanIfRxPduCanIdType: STANDARD_CAN
+        CanIfRxPduReadData: false
+        CanIfRxPduReadNotifyStatus: true
+        CanIfRxPduUserRxIndicationName: PduR_CanIfRxIndication
+        CanIfRxPduUserRxIndicationUL: PDUR
+        CanIfRxPduHrhIdRef: /Ecu_CBO/CanIf/CanIfInitCfg/CanIfInitHohCfg/CanIfHrhCfg0
+        CanIfRxPduRef: 
+  Can:
+    CanConfigSet0:
+      DefinitionRef: CanConfigSet
+      :
+        DefinitionRef: CanHardwareObject
+        CanObjectId: 0
+        CanIdValue: 
+        CanHandleType: FULL
+        CanObjectType: 
+        CanIdType: STANDARD
+        CanControllerRef: /Ecu_CBO/Can/CanConfigSet0/CanController0
+      CanController0:
+        DefinitionRef: CanController
+        CanControllerId: 0
+        CanControllerBaseAddress: 0
+        CanBusoffProcessing: INTERRUPT
+        CanControllerActivation: false
+        CanRxProcessing: INTERRUPT
+        CanTxProcessing: INTERRUPT
+        CanWakeupProcessing: INTERRUPT
+        CanWakeupSupport: false
+        CanWakeupSourceRef: /Ecu_CBO/EcuM/EcuMConfiguration0/EcuMCommonConfiguration0/EcuMWakeupSource0
+        CanControllerDefaultBaudrate: /Ecu_CBO/Can/CanConfigSet0/CanController0/CanControllerBaudrateConfig_0
+        CanControllerBaudrateConfig_0:
+          DefinitionRef: CanControllerBaudrateConfig
+          CanControllerBaudRate: 500
+          CanControllerPropSeg: 0
+          CanControllerSeg1: 0
+          CanControllerSeg2: 255
+          CanControllerSyncJumpWidth: 0
+    CanGeneral:
+      CanIndex: 0
+      CanMainFunctionModePeriod: 0.2
+      CanTimeoutDuration: 0.1
+      CanChangeBaudrateApi: true
+      CanDevErrorDetection: true
+      CanHardwareCancellation: false
+      CanIdenticalIdCancellation: false
+      CanMultiplexedTransmission: false
+      CanVersionInfoApi: true
+      CanCounterRef: /Ecuc/Os/MAIN_HW_COUNTER
+      CanMainFunctionRWPeriods:
+        CanMainFunctionReadPeriod: 0.1
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_cbo/Makefile ecu_education_work_step3/swc-application/1ecu/ecu_cbo/Makefile
--- ecu_education_work_step2/swc-application/1ecu/ecu_cbo/Makefile	2023-08-30 15:38:44.407388700 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_cbo/Makefile	2023-09-11 05:46:34.993476000 +0900
@@ -191,9 +191,9 @@
 ifndef OMIT_OPTIMIZATION
   COPTS := $(COPTS) -O2
 endif
-USER_MACRO = -DTOPPERS_NO_USE_SCHM -DTOPPERS_NO_USE_COMSTACK -DTOPPERS_USE_BODY -DTOPPERS_USE_OPERATION -DTOPPERS_USE_CONTROL -DSIMULATE_ON_ATHRILL -DECU_EDUCATION_USE_ECU_X
+USER_MACRO = -DTOPPERS_USE_BODY -DTOPPERS_USE_OPERATION -DTOPPERS_USE_CONTROL -DSIMULATE_ON_ATHRILL
 CDEFS := $(CDEFS)  $(USER_MACRO)
-INCLUDES := -I. -I$(SRCDIR)/include -I$(SRCDIR)/arch -I$(SRCDIR) $(INCLUDES)
+INCLUDES := -I. -I$(SRCDIR)/include -I$(SRCDIR)/arch -I$(SRCDIR) $(INCLUDES) -I../../../a-comstack/general
 LDFLAGS := $(LDFLAGS) 
 LIBS := $(LIBS) $(CXXLIBS)
 CFLAGS = $(COPTS) $(CDEFS) $(INCLUDES)
@@ -205,7 +205,6 @@
 	CddBodyControl \
 	CddController \
 	CddRcCar \
-	CddSwState \
 	AppLedControl \
 	OperationManager \
 	DriveManager \
@@ -218,9 +217,24 @@
 	driver_if \
 	rcb3 \
 	rlin3x \
-	taub_pwm 
-
-APPLDIR = ../../../driver ../../common/swcs ../../EcuM
+	taub_pwm \
+	Com \
+	CanIf \
+	Can \
+	Can_Irq \
+	Can_Prc \
+	Com_PBcfg \
+	CanIf_PBcfg \
+	CanIf_Lcfg \
+	Can_PBcfg
+APPLDIR = ../../../driver \
+          ../../common/swcs \
+          ../../EcuM \
+          ../../../a-comstack/com \
+          ../../../a-comstack/canif \
+          ../../../a-comstack/can \
+          ../../../a-comstack/can/target/hsbrh850f1l_ccrh \
+          ../../../a-comstack/can/arch/rs_can
 APPL_CFG_INPUT := $(foreach file,$(CFGNAME),$(file).arxml)
 
 APPL_DIR = $(APPLDIR) $(SRCDIR)/library
@@ -231,7 +245,7 @@
 else
   APPL_COBJS = $(APPLNAME:=.o) 
 endif
-APPL_CFLAGS =
+APPL_CFLAGS =-I../../../a-comstack/general -I../../../a-comstack/can/arch/gcc -I../../../a-comstack/stub
 APPL_LIBS =
 ifdef APPLDIR
   INCLUDES := $(INCLUDES) $(foreach dir,$(APPLDIR),-I$(dir))
@@ -625,3 +639,71 @@
 %.o: %.S
 	@echo "*** Default compile rules should not be used."
 	$(CC) -c $(CFLAGS) $<
+
+#
+#  Canジェネレータ
+#
+CAN_PATH = ../../../a-comstack/can
+CAN_ARCH_PATH = $(CAN_PATH)/arch/rs_can
+CAN_TARGET_PATH = $(CAN_PATH)/target/hsbrh850f1k_gcc
+CAN_CFG_TABS = --kernel atk2 --ini-file $(CAN_PATH)/can.ini --api-table $(CAN_PATH)/can.csv
+CAN_CFG_OUT = Can_Cfg.h Can_PBcfg.c
+
+.PHONY: can
+can: $(CAN_CFG_OUT)
+$(CAN_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(CAN_PATH)/can.tf $(CAN_PATH)/can.ini $(CAN_PATH)/can.csv
+	$(CFG) --omit-symbol --pass 2 $(CAN_CFG_TABS) -I $(CAN_PATH) -I $(CAN_ARCH_PATH) -T $(CAN_TARGET_PATH)/Can_Target.tf $(filter %.arxml,$^)
+
+.PHONY: canclean
+canclean:
+	rm -f $(CAN_CFG_OUT)
+
+#
+#  CanIfジェネレータ
+#
+CANIF_PATH = ../../../a-comstack/canif
+CANIF_CFG_TABS = --kernel atk2 --ini-file $(CANIF_PATH)/canif.ini --api-table $(CANIF_PATH)/canif.csv
+CANIF_CFG_OUT = CanIf_Cfg.h CanIf_Lcfg.h CanIf_PBcfg.c CanIf_Lcfg.c
+
+.PHONY: canif
+canif: $(CANIF_CFG_OUT)
+$(CANIF_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(CANIF_PATH)/canif.tf $(CANIF_PATH)/canif.ini $(CANIF_PATH)/canif.csv
+	$(CFG) --omit-symbol --pass 2 $(CANIF_CFG_TABS) -I $(CANIF_PATH) -T $(CANIF_PATH)/canif.tf $(filter %.arxml,$^)
+
+.PHONY: canifclean
+canifclean:
+	rm -f $(CANIF_CFG_OUT)
+
+#
+#  PduRジェネレータ
+#
+PDUR_PATH = ../../../a-comstack/pdur
+PDUR_CFG_TABS = --kernel atk2 --ini-file $(PDUR_PATH)/pdur.ini --api-table $(PDUR_PATH)/pdur.csv
+PDUR_CFG_OUT = PduR_Cfg.h PduR_CanIf.h PduR_Dcm.h PduR_CanTp.h PduR_Com.h
+
+.PHONY: pdur
+pdur: $(PDUR_CFG_OUT)
+$(PDUR_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(PDUR_PATH)/pdur.tf $(PDUR_PATH)/pdur.ini $(PDUR_PATH)/pdur.csv
+	$(CFG) --omit-symbol --pass 2 $(PDUR_CFG_TABS) -I $(PDUR_PATH) -T $(PDUR_PATH)/pdur.tf $(filter %.arxml,$^)
+
+.PHONY: pdurclean
+pdurclean:
+	rm -f $(PDUR_CFG_OUT)
+
+#
+#  Comジェネレータ
+#
+COM_PATH = ../../../a-comstack/com
+COM_CFG_TABS = --kernel atk2 --ini-file $(COM_PATH)/com.ini --api-table $(COM_PATH)/com.csv
+COM_CFG_OUT = Com_Cfg.h Com_PBcfg.c Com_Cbk.h 
+
+.PHONY: com
+com: $(COM_CFG_OUT)
+$(COM_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(COM_PATH)/com.tf $(COM_PATH)/com.ini $(COM_PATH)/com.csv
+	$(CFG) --omit-symbol --pass 2 $(COM_CFG_TABS) -I $(COM_PATH) -T $(COM_PATH)/com.tf $(filter %.arxml,$^)
+
+.PHONY: comclean
+comclean:
+	rm -f $(COM_CFG_OUT)
+
+
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_cbo/device_config_with_can.txt ecu_education_work_step3/swc-application/1ecu/ecu_cbo/device_config_with_can.txt
--- ecu_education_work_step2/swc-application/1ecu/ecu_cbo/device_config_with_can.txt	2023-07-04 04:29:51.851834500 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_cbo/device_config_with_can.txt	2023-09-11 05:46:35.028479500 +0900
@@ -7,11 +7,6 @@
 DEBUG_FUNC_ENABLE_CAN      1
 DEVICE_CONFIG_CAN_FD		100000
 DEBUG_FUNC_CAN_RX_QUEUE_SIZE	256
-DEBUG_FUNC_MROS_NODE_NAME	athrill_test_node
-DEBUG_FUNC_MROS_TOPIC_PUB_NUM   1
-DEBUG_FUNC_MROS_TOPIC_PUB_0 channel0/CAN_IDE0_RTR0_DLC8_0x001
-DEBUG_FUNC_MROS_TOPIC_SUB_NUM   4
-DEBUG_FUNC_MROS_TOPIC_SUB_0 channel0/CAN_IDE0_RTR0_DLC8_0x123
-DEBUG_FUNC_MROS_TOPIC_SUB_1 channel0/CAN_IDE0_RTR0_DLC8_0x122
-DEBUG_FUNC_MROS_TOPIC_SUB_2 channel0/CAN_IDE0_RTR0_DLC8_0x003
-DEBUG_FUNC_MROS_TOPIC_SUB_3 channel0/CAN_IDE0_RTR0_DLC8_0x004
+DEBUG_FUNC_MROS_NODE_NAME	athrill_ecu_cbo
+DEBUG_FUNC_MROS_TOPIC_SUB_NUM   1
+DEBUG_FUNC_MROS_TOPIC_SUB_0 channel0/CAN_IDE0_RTR0_DLC8_0x001
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_cbo/generator.sh ecu_education_work_step3/swc-application/1ecu/ecu_cbo/generator.sh
--- ecu_education_work_step2/swc-application/1ecu/ecu_cbo/generator.sh	2023-07-02 23:20:53.174785700 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_cbo/generator.sh	2023-09-11 05:46:35.035999300 +0900
@@ -3,4 +3,4 @@
 RTE_DIR=../../../a-rtegen
 ARXML_DIR=../../common/arxmls
 APP_NAME=Ecuc_CBO
-$RTE_DIR/bin/bin/rtegen.sh $TARGET_DIR/target_hw_counter.arxml $ARXML_DIR/DataTypes.arxml $ARXML_DIR/BswMD_RLIN3x_RX_ISR.arxml $ARXML_DIR/RcCar.arxml ../System.arxml $APP_NAME.arxml
+$RTE_DIR/bin/bin/rtegen.sh $TARGET_DIR/target_hw_counter.arxml $ARXML_DIR/DataTypes.arxml $ARXML_DIR/BswMD_Rx.arxml ../System_EcuInstance_CBO.arxml $APP_NAME.arxml
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_extract.sh ecu_education_work_step3/swc-application/1ecu/ecu_extract.sh
--- ecu_education_work_step2/swc-application/1ecu/ecu_extract.sh	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_extract.sh	2023-09-11 05:46:35.063515400 +0900
@@ -0,0 +1,2 @@
+RTE_DIR=../../a-rtegen
+ruby $RTE_DIR/ecu_extractor/ecu_extractor.rb ./System.arxml ../common/arxmls/RcCar.arxml 
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_x/Ecuc_X.yaml ecu_education_work_step3/swc-application/1ecu/ecu_x/Ecuc_X.yaml
--- ecu_education_work_step2/swc-application/1ecu/ecu_x/Ecuc_X.yaml	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_x/Ecuc_X.yaml	2023-09-11 05:46:35.072031700 +0900
@@ -0,0 +1,348 @@
+
+Ecu_X:
+  Os:
+    MainApp:
+      DefinitionRef: OsAppMode
+    OsOS:
+      OsStackMonitoring: true
+      OsStatus: EXTENDED
+      OsUseGetServiceId: true
+      OsUseParameterAccess: true
+      OsScalabilityClass: SC1
+      OsHooks:
+        OsErrorHook: true
+        OsProtectionHook: true
+        OsPreTaskHook: false
+        OsPostTaskHook: false
+        OsStartupHook: true
+        OsShutdownHook: false
+      OsHookStack:
+        OsHookStackSize: 1024
+      OsOsStack:
+        OsOsStackSize: 8192
+    ECU_X_TASK_100ms:
+      DefinitionRef: OsTask
+      OsTaskActivation: 1
+      OsTaskPriority: 1
+      OsTaskStackSize: 1024
+      OsTaskSchedule: FULL
+    ECU_INIT_TASK:
+      DefinitionRef: OsTask
+      OsTaskActivation: 1
+      OsTaskPriority: 15
+      OsTaskStackSize: 1024
+      OsTaskSchedule: FULL
+      OsTaskAutostart:
+        OsTaskAppModeRef: /Ecu_X/Os/MainApp
+    ECU_X_SCHM_TASK:
+      DefinitionRef: OsTask
+      OsTaskActivation: 1
+      OsTaskPriority: 2
+      OsTaskStackSize: 1024
+      OsTaskSchedule: FULL
+    OsInclude:
+      OsIncludeFileName:
+        - Can_Target.h
+    RLIN3x_RX_ISR:
+      DefinitionRef: OsIsr
+      OsIsrInterruptPriority: 2
+      OsIsrInterruptNumber: 158 #RLIN32_RX
+      OsIsrCategory: CATEGORY_2
+      OsIsrInterruptSource: ENABLE
+    C2ISR_FOR_CAN_RX0:
+      DefinitionRef: OsIsr
+      OsIsrInterruptPriority: 1
+      OsIsrInterruptNumber: 25 #CAN_RX0
+      OsIsrCategory: CATEGORY_2
+      OsIsrInterruptSource: ENABLE
+    C2ISR_FOR_CAN_TX0:
+      DefinitionRef: OsIsr
+      OsIsrInterruptPriority: 1
+      OsIsrInterruptNumber: 26 #CAN_TX0
+      OsIsrCategory: CATEGORY_2
+      OsIsrInterruptSource: ENABLE
+    ECU_X_ALARM_100ms:
+      DefinitionRef: OsAlarm
+      OsAlarmCounterRef: /Ecuc/Os/MAIN_HW_COUNTER
+      OsAlarmAction:
+        OsAlarmActivateTask:
+          OsAlarmActivateTaskRef: /Ecu_X/Os/ECU_X_TASK_100ms
+    ECU_X_ALARM_20ms_SCHM:
+      DefinitionRef: OsAlarm
+      OsAlarmCounterRef: /Ecuc/Os/MAIN_HW_COUNTER
+      OsAlarmAction:
+        OsAlarmActivateTask:
+          OsAlarmActivateTaskRef: /Ecu_X/Os/ECU_X_SCHM_TASK
+  Rte:
+    RteGeneration:
+      RteCodeVendorId: 65
+      RteCalibrationSupport: NONE
+      RteDevErrorDetect: true
+      RteDevErrorDetectUninit: true
+      RteGenerationMode: COMPATIBILITY_MODE
+      RteOptimizationMode: MEMORY
+      RteValueRangeCheckEnabled: false
+      RteVfbTraceEnabled: false
+    RteInitializationBehavior:
+      RteInitializationStrategy: RTE_INITIALIZATION_STRATEGY_AT_RTE_START_AND_PARTITION_RESTART
+      RteSectionInitializationPolicy: VALUE
+    SWCI_CddSwState:
+      DefinitionRef: RteSwComponentInstance
+      RteSoftwareComponentInstanceRef: /RcCar/Composition_RcCar/CddSwState
+      IB_CSS_TimingEvent_SwState_500ms:
+        DefinitionRef: RteEventToTaskMapping
+        RteActivationOffset: 0.0
+        RtePositionInTask: 9
+        RteEventRef: /RcCar/CddSwState/IB_CSS/SwState_timingEvent
+        RteMappedToTaskRef: /Ecu_X/Os/ECU_X_TASK_100ms
+        RteUsedOsAlarmRef: /Ecu_X/Os/ECU_X_ALARM_100ms
+    RteOsInteraction:
+      RteUsedOsActivation1:
+        DefinitionRef: RteUsedOsActivation
+        RteExpectedActivationOffset: 0.0
+        RteExpectedTickDuration: 0.1
+        RteActivationOsAlarmRef: /Ecu_X/Os/ECU_X_ALARM_100ms
+      RteUsedOsActivation2:
+        DefinitionRef: RteUsedOsActivation
+        RteExpectedActivationOffset: 0.0
+        RteExpectedTickDuration: 0.02
+        RteActivationOsAlarmRef: /Ecu_X/Os/ECU_X_ALARM_20ms_SCHM
+    RteBswModuleInstance_com:
+      DefinitionRef: RteBswModuleInstance
+      RteBswImplementationRef: /BSWScheduler/BswImplementations/bswImplementation0
+      RteBswExclusiveAreaImpl_exclusiveArea0:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/Reentrant_0
+      RteBswExclusiveAreaImpl_exclusiveArea1:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/Reentrant_1
+      RteBswExclusiveAreaImpl_exclusiveArea2:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/Reentrant_2
+      RteBswEventToTaskMapping_timingEvent1:
+        DefinitionRef: RteBswEventToTaskMapping
+        RteBswPositionInTask: 0
+        RteBswActivationOffset: 0.02
+        RteBswEventRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/TimingEvent_MainFunctionRx
+        RteBswMappedToTaskRef: /Ecu_X/Os/ECU_X_SCHM_TASK
+        RteBswUsedOsAlarmRef: /Ecu_X/Os/ECU_X_ALARM_20ms_SCHM
+      RteBswEventToTaskMapping_timingEvent2:
+        DefinitionRef: RteBswEventToTaskMapping
+        RteBswPositionInTask: 1
+        RteBswActivationOffset: 0.02
+        RteBswEventRef: /BSWScheduler/BswModuleDescriptions/Com/InternalBehavior_0/TimingEvent_MainFunctionTx
+        RteBswMappedToTaskRef: /Ecu_X/Os/ECU_X_SCHM_TASK
+        RteBswUsedOsAlarmRef: /Ecu_X/Os/ECU_X_ALARM_20ms_SCHM
+    RteBswModuleInstance_can:
+      DefinitionRef: RteBswModuleInstance
+      RteBswImplementationRef: /BSWScheduler/BswImplementations/bswImplementation1
+      RteBswExclusiveAreaImpl_exclusiveArea3:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/Can/InternalBehavior_0/Reentrant
+    RteBswModuleInstance_canif:
+      DefinitionRef: RteBswModuleInstance
+      RteBswImplementationRef: /BSWScheduler/BswImplementations/bswImplementation2
+      RteBswExclusiveAreaImpl_exclusiveArea4:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/CanIf/InternalBehavior_0/Reentrant_0
+      RteBswExclusiveAreaImpl_exclusiveArea5:
+        DefinitionRef: RteBswExclusiveAreaImpl
+        RteExclusiveAreaImplMechanism: NONE
+        RteBswExclusiveAreaRef: /BSWScheduler/BswModuleDescriptions/CanIf/InternalBehavior_0/Reentrant_1
+    RteBswGeneral:
+      RteSchMVersionInfoApi: true
+      RteUseComShadowSignalApi: false
+  EcuC:
+    EcucPduCollection:
+      PduIdTypeEnum: UINT16
+      PduLengthTypeEnum: UINT32
+      Pdu0:
+        DefinitionRef: Pdu
+        PduLength: 8
+  EcuM:
+    EcuMConfiguration0:
+      DefinitionRef: EcuMConfiguration
+      EcuMCommonConfiguration0:
+        DefinitionRef: EcuMCommonConfiguration
+        EcuMWakeupSource0:
+          DefinitionRef: EcuMWakeupSource
+          EcuMWakeupSourceId: 0
+          EcuMWakeupSourcePolling: false
+  Com:
+    ComConfig:
+      ComConfigurationId: 0
+      ComSignal0:
+        DefinitionRef: ComSignal
+        ComHandleId: 0
+        ComBitPosition: 0
+        ComBitSize: 8
+        ComSignalType: UINT8
+        ComTransferProperty: TRIGGERED_ON_CHANGE
+        ComSignalEndianness: LITTLE_ENDIAN
+        ComSystemTemplateSystemSignalRef: /Network/Ecu_X/ISignalIPdu_SwStateOut/ISignalToIPduMapping_SwStateOut_dsw
+      ComSignal1:
+        DefinitionRef: ComSignal
+        ComHandleId: 1
+        ComBitPosition: 8
+        ComBitSize: 8
+        ComSignalType: UINT8
+        ComSignalEndianness: LITTLE_ENDIAN
+        ComTransferProperty: TRIGGERED_ON_CHANGE
+        ComSystemTemplateSystemSignalRef: /Network/Ecu_X/ISignalIPdu_SwStateOut/ISignalToIPduMapping_SwStateOut_psw
+      ComIPdu0:
+        DefinitionRef: ComIPdu
+        ComIPduHandleId: 0
+        ComIPduDirection: SEND
+        ComIPduSignalProcessing: IMMEDIATE
+        ComIPduType: NORMAL
+        ComIPduGroupRef: /Ecu_X/Com/ComConfig/ComIPduGroup0
+        ComIPduSignalRef:
+          - /Ecu_X/Com/ComConfig/ComSignal0
+          - /Ecu_X/Com/ComConfig/ComSignal1
+        ComPduIdRef: /Ecu_X/EcuC/EcucPduCollection/Pdu0
+        ComTxIPdu:
+          ComTxIPduUnusedAreasDefault: 0
+          ComTxModeTrue:
+            ComTxMode:
+              ComTxModeTimeOffset: 0.02
+              ComTxModeTimePeriod: 0.02
+              ComTxModeMode: PERIODIC
+      ComIPduGroup0:
+        DefinitionRef: ComIPduGroup
+        ComIPduGroupHandleId: 0
+      ComTimeBase:
+        ComGwTimeBase: 0.02
+        ComRxTimeBase: 0.02
+        ComTxTimeBase: 0.02
+    ComGeneral:
+      ComSupportedIPduGroups: 1
+      ComVersionInfoApi: false
+      ComConfigurationUseDet: true
+  PduR:
+    PduRGeneral:
+      PduRZeroCostOperation: true
+      PduRDevErrorDetect: false
+      PduRVersionInfoApi: false
+    PduRRoutingTables:
+      PduRConfigurationId: 0
+      PduRRoutingTable:
+        PduRIsMinimumRouting: false
+        PduRRoutingPath_0:
+          DefinitionRef: PduRRoutingPath
+          PduRSrcPdu0:
+            DefinitionRef: PduRSrcPdu
+            PduRSourcePduHandleId: 0
+            PduRSrcPduRef: /Ecu_X/EcuC/EcucPduCollection/Pdu0
+          PduRDestPdu10:
+            DefinitionRef: PduRDestPdu
+            PduRDestPduHandleId: 0
+            PduRDestPduRef: /Ecu_X/EcuC/EcucPduCollection/Pdu0
+  CanIf:
+    CanIfPublicCfg:
+      CanIfPublicNumberOfCanHwUnits: 1
+      CanIfPublicCancelTransmitSupport: true
+      CanIfPublicChangeBaudrateSupport: true
+      CanIfPublicDevErrorDetect: true
+      CanIfPublicHandleTypeEnum: UINT8
+      CanIfPublicMultipleDrvSupport: false
+      CanIfPublicPnSupport: false
+      CanIfPublicReadRxPduDataApi: true
+      CanIfPublicReadRxPduNotifyStatusApi: true
+      CanIfPublicReadTxPduNotifyStatusApi: true
+      CanIfPublicSetDynamicTxIdApi: false
+      CanIfPublicTxBuffering: false
+      CanIfPublicTxConfirmPollingSupport: true
+      CanIfPublicVersionInfoApi: false
+      CanIfPublicWakeupCheckValidSupport: true
+    CanIfPrivateCfg:
+      CanIfPrivateDlcCheck: true
+      CanIfSupportTTCAN: false
+    CanIfDispatchCfg:
+      CanIfDispatchUserCtrlBusOffName: CanSM_ControllerBusOff
+      CanIfDispatchUserCtrlBusOffUL: CAN_SM
+      CanIfDispatchUserCtrlModeIndicationName: CanSM_ControllerModeIndication
+      CanIfDispatchUserCtrlModeIndicationUL: CAN_SM
+      CanIfDispatchUserValidateWakeupEventName: EcuM_ValidateWakeupEvent
+      CanIfDispatchUserValidateWakeupEventUL: ECUM
+    CanIfCtrlDrvCfg:
+      CanIfCtrlDrvTxCancellation: false
+      CanIfCtrlDrvInitHohConfigRef: /Ecu_X/CanIf/CanIfInitCfg/CanIfInitHohCfg
+      CanIfCtrlDrvNameRef: /Ecu_X/Can/CanGeneral
+      CanIfCtrlCfg0:
+        DefinitionRef: CanIfCtrlCfg
+        CanIfCtrlId: 0
+        CanIfCtrlWakeupSupport: true
+        CanIfCtrlCanCtrlRef: /Ecu_X/Can/CanConfigSet0/CanController0
+    CanIfInitCfg:
+      CanIfInitCfgSet: main_canif_config
+      CanIfBufferCfg0:
+        DefinitionRef: CanIfBufferCfg
+        CanIfBufferSize: 0
+        CanIfBufferHthRef: /Ecu_X/CanIf/CanIfInitCfg/CanIfInitHohCfg/CanIfHthCfg0
+      CanIfInitHohCfg:
+        CanIfInitRefCfgSet: /Ecu_X/Can/CanConfigSet0
+        CanIfHthCfg0:
+          DefinitionRef: CanIfHthCfg
+          CanIfHthCanCtrlIdRef: /Ecu_X/CanIf/CanIfCtrlDrvCfg/CanIfCtrlCfg0
+          CanIfHthIdSymRef: /Ecu_X/Can/CanConfigSet0/CanHardwareObject0
+      CanIfTxPduCfg0:
+        DefinitionRef: CanIfTxPduCfg
+        CanIfTxPduCanId: 1
+        CanIfTxPduDlc: 8
+        CanIfTxPduId: 0
+        CanIfTxPduCanIdType: STANDARD_CAN
+        CanIfTxPduPnFilterPdu: false
+        CanIfTxPduReadNotifyStatus: true
+        CanIfTxPduType: STATIC
+        CanIfTxPduUserTxConfirmationName: PduR_CanIfTxConfirmation
+        CanIfTxPduUserTxConfirmationUL: PDUR
+        CanIfTxPduBufferRef: /Ecu_X/CanIf/CanIfInitCfg/CanIfBufferCfg0
+        CanIfTxPduRef: /Ecu_X/EcuC/EcucPduCollection/Pdu0
+  Can:
+    CanConfigSet0:
+      DefinitionRef: CanConfigSet
+      CanHardwareObject0:
+        DefinitionRef: CanHardwareObject
+        CanObjectId: 0
+        CanIdValue: 1
+        CanHandleType: FULL
+        CanObjectType: TRANSMIT
+        CanIdType: STANDARD
+        CanControllerRef: /Ecu_X/Can/CanConfigSet0/CanController0
+      CanController0:
+        DefinitionRef: CanController
+        CanControllerId: 0
+        CanControllerBaseAddress: 0
+        CanBusoffProcessing: INTERRUPT
+        CanControllerActivation: false
+        CanRxProcessing: INTERRUPT
+        CanTxProcessing: INTERRUPT
+        CanWakeupProcessing: INTERRUPT
+        CanWakeupSupport: false
+        CanWakeupSourceRef: /Ecu_X/EcuM/EcuMConfiguration0/EcuMCommonConfiguration0/EcuMWakeupSource0
+        CanControllerDefaultBaudrate: /Ecu_X/Can/CanConfigSet0/CanController0/CanControllerBaudrateConfig_0
+        CanControllerBaudrateConfig_0:
+          DefinitionRef: CanControllerBaudrateConfig
+          CanControllerBaudRate: 500
+          CanControllerPropSeg: 0
+          CanControllerSeg1: 0
+          CanControllerSeg2: 255
+          CanControllerSyncJumpWidth: 0
+    CanGeneral:
+      CanIndex: 0
+      CanMainFunctionModePeriod: 0.2
+      CanTimeoutDuration: 0.1
+      CanChangeBaudrateApi: true
+      CanDevErrorDetection: true
+      CanHardwareCancellation: false
+      CanIdenticalIdCancellation: false
+      CanMultiplexedTransmission: false
+      CanVersionInfoApi: true
+      CanCounterRef: /Ecuc/Os/MAIN_HW_COUNTER
+      CanMainFunctionRWPeriods:
+        CanMainFunctionReadPeriod: 0.1
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_x/Makefile ecu_education_work_step3/swc-application/1ecu/ecu_x/Makefile
--- ecu_education_work_step2/swc-application/1ecu/ecu_x/Makefile	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_x/Makefile	2023-09-11 05:46:35.079684900 +0900
@@ -0,0 +1,703 @@
+#
+#  TOPPERS ATK2
+#      Toyohashi Open Platform for Embedded Real-Time Systems
+#      Automotive Kernel Version 2
+#
+#  Copyright (C) 2000-2003 by Embedded and Real-Time Systems Laboratory
+#                              Toyohashi Univ. of Technology, JAPAN
+#  Copyright (C) 2006-2017 by Center for Embedded Computing Systems
+#              Graduate School of Information Science, Nagoya Univ., JAPAN
+#  Copyright (C) 2011-2017 by FUJI SOFT INCORPORATED, JAPAN
+#  Copyright (C) 2011-2013 by Spansion LLC, USA
+#  Copyright (C) 2011-2017 by NEC Communication Systems, Ltd., JAPAN
+#  Copyright (C) 2011-2016 by Panasonic Advanced Technology Development Co., Ltd., JAPAN
+#  Copyright (C) 2011-2014 by Renesas Electronics Corporation, JAPAN
+#  Copyright (C) 2011-2016 by Sunny Giken Inc., JAPAN
+#  Copyright (C) 2011-2017 by TOSHIBA CORPORATION, JAPAN
+#  Copyright (C) 2011-2017 by Witz Corporation
+#  Copyright (C) 2014-2016 by AISIN COMCRUISE Co., Ltd., JAPAN
+#  Copyright (C) 2014-2016 by eSOL Co.,Ltd., JAPAN
+#  Copyright (C) 2014-2017 by SCSK Corporation, JAPAN
+#  Copyright (C) 2015-2017 by SUZUKI MOTOR CORPORATION
+#
+#  上記著作権者は，以下の(1)〜(4)の条件を満たす場合に限り，本ソフトウェ
+#  ア（本ソフトウェアを改変したものを含む．以下同じ）を使用・複製・改
+#  変・再配布（以下，利用と呼ぶ）することを無償で許諾する．
+#  (1) 本ソフトウェアをソースコードの形で利用する場合には，上記の著作
+#      権表示，この利用条件および下記の無保証規定が，そのままの形でソー
+#      スコード中に含まれていること．
+#  (2) 本ソフトウェアを，ライブラリ形式など，他のソフトウェア開発に使
+#      用できる形で再配布する場合には，再配布に伴うドキュメント（利用
+#      者マニュアルなど）に，上記の著作権表示，この利用条件および下記
+#      の無保証規定を掲載すること．
+#  (3) 本ソフトウェアを，機器に組み込むなど，他のソフトウェア開発に使
+#      用できない形で再配布する場合には，次のいずれかの条件を満たすこ
+#      と．
+#    (a) 再配布に伴うドキュメント（利用者マニュアルなど）に，上記の著
+#        作権表示，この利用条件および下記の無保証規定を掲載すること．
+#    (b) 再配布の形態を，別に定める方法によって，TOPPERSプロジェクトに
+#        報告すること．
+#  (4) 本ソフトウェアの利用により直接的または間接的に生じるいかなる損
+#      害からも，上記著作権者およびTOPPERSプロジェクトを免責すること．
+#      また，本ソフトウェアのユーザまたはエンドユーザからのいかなる理
+#      由に基づく請求からも，上記著作権者およびTOPPERSプロジェクトを
+#      免責すること．
+#
+#  本ソフトウェアは，AUTOSAR（AUTomotive Open System ARchitecture）仕
+#  様に基づいている．上記の許諾は，AUTOSARの知的財産権を許諾するもので
+#  はない．AUTOSARは，AUTOSAR仕様に基づいたソフトウェアを商用目的で利
+#  用する者に対して，AUTOSARパートナーになることを求めている．
+#
+#  本ソフトウェアは，無保証で提供されているものである．上記著作権者お
+#  よびTOPPERSプロジェクトは，本ソフトウェアに関して，特定の使用目的
+#  に対する適合性も含めて，いかなる保証も行わない．また，本ソフトウェ
+#  アの利用により直接的または間接的に生じたいかなる損害に関しても，そ
+#  の責任を負わない．
+#
+#  $Id: Makefile 727 2017-01-23 09:27:59Z witz-itoyo $
+#
+
+#
+#  ターゲットの指定（Makefile.targetで上書きされるのを防ぐため）
+#
+all:
+
+#
+#  ターゲット略称の定義
+#
+TARGET = hsbrh850f1k_gcc
+
+#
+#  プログラミング言語の定義
+#
+SRCLANG = c
+ifeq ($(SRCLANG),c++)
+  USE_CXX = true
+  CXXLIBS = -lstdc++ -lm -lc
+  CXXRTS = cxxrt.o newlibrt.o
+endif
+
+#
+#  ソースファイルのディレクトリの定義
+#
+SRCDIR = ../../../atk2-sc1
+
+#
+#  オブジェクトファイル名の拡張子の設定
+#
+OBJEXT = 
+
+#
+#  実行環境の定義（ターゲット依存に上書きされる場合がある）
+#
+DBGENV := 
+
+#
+#  カーネルライブラリ（libkernel.a）のディレクトリ名
+#  （カーネルライブラリもmake対象にする時は，空に定義する）
+#
+KERNEL_LIB = 
+
+#
+#  カーネルを関数単位でコンパイルするかどうかの定義
+#
+KERNEL_FUNCOBJS = 
+
+#
+#  トレースログを取得するかどうかの定義
+#
+ENABLE_TRACE = 
+
+#
+#  システムタイマ(タイマドライバ)を使用するかどうかの定義
+#
+ENABLE_SYS_TIMER = 
+
+#
+#  ハードウェアカウンタを無効にするかどうかの定義
+#
+OMIT_HW_COUNTER = 
+
+#
+#  トレースログに関する設定
+#
+ifeq ($(ENABLE_TRACE),true)
+      ENABLE_SYS_TIMER = true
+      CDEFS := $(CDEFS) -DTOPPERS_ENABLE_TRACE
+      KERNEL_DIR := $(KERNEL_DIR) $(SRCDIR)/arch/logtrace
+      KERNEL_COBJS := $(KERNEL_COBJS) trace_config.o trace_dump.o
+endif
+
+#  コンフィギュレーションファイルに関する定義
+#
+CFGNAME = Ecuc_X
+CFGNAME := $(CFGNAME) target_serial
+ifdef ENABLE_SYS_TIMER
+  CFGNAME := $(CFGNAME) target_timer
+endif
+ifndef OMIT_HW_COUNTER
+  CFGNAME := $(CFGNAME) target_hw_counter
+endif
+
+#
+#  ユーティリティプログラムの名称
+#
+PERL = /usr/bin/perl
+CFG = $(SRCDIR)/cfg/cfg/cfg
+
+#
+#  オブジェクトファイル名の定義
+#
+OBJNAME = atk2-sc1
+ifdef OBJEXT
+  OBJFILE = $(OBJNAME).$(OBJEXT)
+  CFG1_OUT = cfg1_out.$(OBJEXT)
+else
+  OBJFILE = $(OBJNAME)
+  CFG1_OUT = cfg1_out
+endif
+
+#
+#  ターゲット依存部のディレクトリの定義
+#
+TARGETDIR = $(SRCDIR)/target/$(TARGET)
+
+#
+#  ターゲット依存の定義のインクルード
+#
+include $(TARGETDIR)/Makefile.target
+
+#
+#  ジェネレータ関係の変数の定義
+#
+PASS2_TF = $(SRCDIR)/kernel/kernel.tf
+DEF_TABS = $(SRCDIR)/kernel/kernel.csv
+CFG_KERNEL = atk2
+CFG_TABS := --ini-file $(SRCDIR)/kernel/kernel.ini $(CFG_TABS)
+
+CFG_TABS := --api-table $(DEF_TABS) \
+			--cfg1-def-table $(SRCDIR)/kernel/kernel_def.csv $(CFG_TABS)
+CFG_OBJS := Os_Lcfg.o $(CFG_OBJS)
+CFG2_OUT := Os_Lcfg.c Os_Lcfg.h Os_Cfg.h $(CFG2_OUT)
+ALL_CFG_OBJS := $(CFG_OBJS) cfg1_out.o
+
+#
+#  共通コンパイルオプションの定義
+#
+COPTS := $(COPTS) -g 
+ifndef OMIT_WARNING_ALL
+  COPTS := $(COPTS) -Wall -Wno-unused-label -Wpointer-arith
+endif
+ifndef OMIT_OPTIMIZATION
+  COPTS := $(COPTS) -O2
+endif
+USER_MACRO = -DECU_EDUCATION_USE_ECU_X -DSIMULATE_ON_ATHRILL
+CDEFS := $(CDEFS)  $(USER_MACRO)
+INCLUDES := -I. -I$(SRCDIR)/include -I$(SRCDIR)/arch -I$(SRCDIR) $(INCLUDES) -I../../../a-comstack/general
+LDFLAGS := $(LDFLAGS) 
+LIBS := $(LIBS) $(CXXLIBS)
+CFLAGS = $(COPTS) $(CDEFS) $(INCLUDES)
+
+#
+#  アプリケーションプログラムに関する定義
+#
+APPLNAME = CddSwState \
+	Os_Hook \
+	C_Init_Code \
+	EcuM \
+	EcuM_StartupTask \
+	Rte \
+	hsbrh850f1l_device \
+	driver_if \
+	rlin3x \
+	taub_pwm \
+	Com \
+	CanIf \
+	Can \
+	Can_Irq \
+	Can_Prc \
+	Com_PBcfg \
+	CanIf_PBcfg \
+	CanIf_Lcfg \
+	Can_PBcfg
+APPLDIR = ../../../driver \
+          ../../common/swcs \
+          ../../EcuM \
+          ../../../a-comstack/com \
+          ../../../a-comstack/canif \
+          ../../../a-comstack/can \
+          ../../../a-comstack/can/target/hsbrh850f1l_ccrh \
+          ../../../a-comstack/can/arch/rs_can
+APPL_CFG_INPUT := $(foreach file,$(CFGNAME),$(file).arxml)
+
+APPL_DIR = $(APPLDIR) $(SRCDIR)/library
+APPL_ASMOBJS =
+ifdef USE_CXX
+  APPL_CXXOBJS = $(APPLNAME).o 
+  APPL_COBJS =
+else
+  APPL_COBJS = $(APPLNAME:=.o) 
+endif
+APPL_CFLAGS =-I../../../a-comstack/general -I../../../a-comstack/can/arch/gcc -I../../../a-comstack/stub
+APPL_LIBS =
+ifdef APPLDIR
+  INCLUDES := $(INCLUDES) $(foreach dir,$(APPLDIR),-I$(dir))
+endif
+
+
+#
+#  システムモジュールに関する定義
+#
+SYSMOD_DIR := $(SYSMOD_DIR) $(SRCDIR)/sysmod $(SRCDIR)/library
+SYSMOD_ASMOBJS := $(SYSMOD_ASMOBJS)
+SYSMOD_COBJS := $(SYSMOD_COBJS) banner.o syslog.o serial.o \
+				log_output.o vasyslog.o t_perror.o strerror.o \
+				$(CXXRTS)
+SYSMOD_CFLAGS := $(SYSMOD_CFLAGS)
+SYSMOD_LIBS := $(SYSMOD_LIBS)
+INCLUDES := $(INCLUDES)
+
+#
+#  カーネルに関する定義
+#
+#  KERNEL_ASMOBJS: カーネルライブラリに含める，ソースがアセンブリ言語の
+#				   オブジェクトファイル．
+#  KERNEL_COBJS: カーネルのライブラリに含める，ソースがC言語で，ソース
+#				 ファイルと1対1に対応するオブジェクトファイル．
+#  KERNEL_LCSRCS: カーネルのライブラリに含めるC言語のソースファイルで，
+#				  1つのソースファイルから複数のオブジェクトファイルを生
+#				  成するもの．
+#  KERNEL_LCOBJS: 上のソースファイルから生成されるオブジェクトファイル．
+#  KERNEL_AUX_COBJS: ロードモジュールに含めないが，カーネルのソースファ
+#					 イルと同じオプションを適用してコンパイルすべき，ソー
+#					 スがC言語のオブジェクトファイル．
+#
+KERNEL_DIR := $(KERNEL_DIR) $(SRCDIR)/kernel
+KERNEL_ASMOBJS := $(KERNEL_ASMOBJS)
+KERNEL_COBJS := $(KERNEL_COBJS)
+KERNEL_CFLAGS := $(KERNEL_CFLAGS) -I$(SRCDIR)/kernel
+ifdef OMIT_MAKEOFFSET
+  OFFSET_H =
+else
+  OFFSET_H = offset.h
+endif
+
+
+#
+#  ターゲットファイル（複数を同時に選択してはならない）
+#
+all: $(OBJFILE)
+#all: $(OBJNAME).bin
+#all: $(OBJNAME).srec
+
+##### 以下は編集しないこと #####
+
+#
+#  環境に依存するコンパイルオプションの定義
+#
+ifdef DBGENV
+  CDEFS := $(CDEFS) -D$(DBGENV)
+endif
+
+#
+#  カーネルのファイル構成の定義
+#
+include $(SRCDIR)/kernel/Makefile.kernel
+ifdef KERNEL_FUNCOBJS
+  KERNEL_LCSRCS := $(KERNEL_FCSRCS)
+  KERNEL_LCOBJS := $(foreach file,$(KERNEL_FCSRCS),$($(file:.c=)))
+else
+  KERNEL_CFLAGS := -DALLFUNC $(KERNEL_CFLAGS)
+  KERNEL_COBJS := $(KERNEL_COBJS) \
+					$(foreach file,$(KERNEL_FCSRCS),$(file:.c=.o))
+endif
+
+#
+#  ソースファイルのあるディレクトリに関する定義
+#
+vpath %.c $(KERNEL_DIR) $(SYSMOD_DIR) $(APPL_DIR)
+vpath %.S $(KERNEL_DIR) $(SYSMOD_DIR) $(APPL_DIR)
+vpath %.arxml $(KERNEL_DIR) $(SYSMOD_DIR) $(APPL_DIR)
+
+#
+#  コンパイルのための変数の定義
+#
+KERNEL_LIB_OBJS = $(KERNEL_ASMOBJS) $(KERNEL_COBJS) $(KERNEL_LCOBJS)
+SYSMOD_OBJS = $(SYSMOD_ASMOBJS) $(SYSMOD_COBJS)
+APPL_OBJS = $(APPL_ASMOBJS) $(APPL_COBJS) $(APPL_CXXOBJS)
+ALL_OBJS = $(START_OBJS) $(APPL_OBJS) $(SYSMOD_OBJS) $(CFG_OBJS) \
+											$(END_OBJS) $(HIDDEN_OBJS)
+ifdef KERNEL_LIB
+  ALL_LIBS = $(APPL_LIBS) $(SYSMOD_LIBS) -lkernel $(LIBS)
+  LIBS_DEP = $(filter %.a,$(ALL_LIBS)) $(KERNEL_LIB)/libkernel.a
+  LDFLAGS := $(LDFLAGS) -L$(KERNEL_LIB)
+else
+  ALL_LIBS = $(APPL_LIBS) $(SYSMOD_LIBS) libkernel.a $(LIBS)
+  LIBS_DEP = $(filter %.a,$(ALL_LIBS))
+endif
+
+ifdef TEXT_START_ADDRESS
+  LDFLAGS := $(LDFLAGS) -Wl,-Ttext,$(TEXT_START_ADDRESS)
+endif
+ifdef DATA_START_ADDRESS
+  LDFLAGS := $(LDFLAGS) -Wl,-Tdata,$(DATA_START_ADDRESS)
+endif
+ifdef LDSCRIPT
+  LDFLAGS := $(LDFLAGS) -T $(LDSCRIPT)
+endif
+
+#
+#  オフセットファイル（offset.h）の生成規則
+#
+offset.h: $(APPL_CFG_INPUT) Os_Lcfg.timestamp $(SRCDIR)/kernel/genoffset.tf
+	$(CFG) --pass 3 --kernel $(CFG_KERNEL) $(INCLUDES) \
+				--rom-image cfg1_out.srec --symbol-table cfg1_out.syms \
+				-T $(OFFSET_TF) $(CFG_TABS) $(filter %.arxml,$^)
+
+#
+#  カーネルのコンフィギュレーションファイルの生成
+#
+cfg1_out.c: $(APPL_CFG_INPUT) $(SRCDIR)/kernel/kernel_def.csv
+	$(CFG) --pass 1 --kernel $(CFG_KERNEL) $(INCLUDES) $(CFG_TABS) $(filter %.arxml,$^)
+
+$(CFG2_OUT): Os_Lcfg.timestamp
+Os_Lcfg.timestamp: $(APPL_CFG_INPUT) $(START_OBJS) cfg1_out.o $(END_OBJS) $(HIDDEN_OBJS) $(PASS2_TF) $(DEF_TABS)
+	$(LINK) $(CFLAGS) $(LDFLAGS) $(CFG1_OUT_LDFLAGS) -o $(CFG1_OUT) \
+						$(START_OBJS) cfg1_out.o $(END_OBJS)
+	$(NM) -n $(CFG1_OUT) > cfg1_out.syms
+	$(OBJCOPY) -O srec -S $(CFG1_OUT) cfg1_out.srec
+	$(CFG) --pass 2 --kernel $(CFG_KERNEL) $(INCLUDES) \
+				-T $(TARGETDIR)/target.tf $(CFG_TABS) $(filter %.arxml,$^)
+
+#  既存Os_Cfg.hとpass2で生成したOs_Cfg_tmp.hとの差分がある場合Os_Cfg.hを上書き
+	if ! cmp Os_Cfg.h Os_Cfg_tmp.h >/dev/null 2>&1 ; then \
+		mv Os_Cfg_tmp.h Os_Cfg.h ;\
+	else \
+		rm Os_Cfg_tmp.h ;\
+	fi
+ifdef KERNEL_LIB
+	if ! cmp Os_Cfg.h $(KERNEL_LIB)/Os_Cfg.h >/dev/null 2>&1 ; then \
+		echo Config files Os_Cfg.h and $(KERNEL_LIB)/Os_Cfg.h differ! ;\
+		diff -c Os_Cfg.h $(KERNEL_LIB)/Os_Cfg.h ;\
+		false;\
+	fi
+endif
+
+	touch -r Os_Lcfg.c Os_Lcfg.timestamp
+
+#
+#  カーネルライブラリファイルの生成
+#
+libkernel.a: $(OFFSET_H) $(KERNEL_LIB_OBJS)
+	rm -f libkernel.a
+	$(AR) -rcs libkernel.a $(KERNEL_LIB_OBJS)
+	$(RANLIB) libkernel.a
+
+#
+#  特別な依存関係の定義
+#
+banner.o: Os_Lcfg.timestamp $(filter-out banner.o,$(ALL_OBJS)) $(LIBS_DEP)
+
+#
+#  全体のリンク
+#
+$(OBJFILE): $(APPL_CFG_INPUT) Os_Lcfg.timestamp $(ALL_OBJS) $(LIBS_DEP)
+	$(LINK) $(CFLAGS) $(LDFLAGS) -o $(OBJFILE) $(START_OBJS) \
+			-Wl,-Map,$(OBJNAME).map \
+			$(APPL_OBJS) $(SYSMOD_OBJS) $(CFG_OBJS) $(ALL_LIBS) $(END_OBJS)
+	$(NM) -n $(OBJFILE) > $(OBJNAME).syms
+	$(OBJCOPY) -O srec -S $(OBJFILE) $(OBJNAME).srec
+	$(CFG) --pass 3 --kernel $(CFG_KERNEL) $(INCLUDES) \
+				--rom-image $(OBJNAME).srec --symbol-table $(OBJNAME).syms \
+				-T $(TARGETDIR)/target_check.tf $(CFG_TABS) $(filter %.arxml,$^)
+	$(OBJDUMP) -d $(OBJFILE) > $(OBJNAME).dump
+
+#
+#  バイナリファイルの生成
+#
+$(OBJNAME).bin: $(OBJFILE)
+	$(OBJCOPY) -O binary -S $(OBJFILE) $(OBJNAME).bin
+
+#
+#  Sレコードファイルの生成
+#
+$(OBJNAME).srec: $(OBJFILE)
+	$(OBJCOPY) -O srec -S $(OBJFILE) $(OBJNAME).srec
+
+#
+#  コンパイル結果の消去
+#
+.PHONY: clean
+clean:
+	rm -f \#* *~ *.o $(CLEAN_FILES)
+	rm -f $(OBJFILE) $(OBJNAME).syms $(OBJNAME).srec $(OBJNAME).bin $(OBJNAME).map
+	rm -f Os_Lcfg.timestamp $(CFG2_OUT)
+	rm -f cfg2_out.tf
+	rm -f cfg1_out.c $(CFG1_OUT) cfg1_out.syms cfg1_out.srec
+	rm -f $(OBJNAME).dump
+	rm -f kernel.res
+ifndef KERNEL_LIB
+	rm -f libkernel.a
+endif
+	rm -f offset.h
+
+.PHONY: cleankernel
+cleankernel:
+	rm -rf $(KERNEL_LIB_OBJS)
+	rm -f offset.h
+
+.PHONY: cleandep
+cleandep:
+	if ! [ -f Makefile.depend ]; then \
+		rm -f Os_Lcfg.timestamp $(CFG2_OUT); \
+		rm -f cfg1_out.c cfg1_out.o $(CFG1_OUT) cfg1_out.syms cfg1_out.srec; \
+		rm -f offset.h; \
+	fi
+	rm -f Makefile.depend
+
+.PHONY: realclean
+realclean: cleandep clean
+	rm -f $(REALCLEAN_FILES)
+
+#
+#  ジェネレータが生成したファイルのコンパイルルールと依存関係作成
+#  ルールの定義
+#
+#  ジェネレータが生成したファイルは，アプリケーションプログラム用，シス
+#  テムサービス用，カーネル用のすべてのオプションを付けてコンパイルする．
+#
+OS_LCFG_CFLAGS = $(APPL_CFLAGS) $(SYSMOD_CFLAGS) $(KERNEL_CFLAGS)
+
+$(ALL_CFG_OBJS): %.o: %.c
+	$(CC) -c $(CFLAGS) $(OS_LCFG_CFLAGS) $<
+
+$(ALL_CFG_OBJS:.o=.s): %.s: %.c
+	$(CC) -S $(CFLAGS) $(OS_LCFG_CFLAGS) $<
+
+$(ALL_CFG_OBJS:.o=.d): %.d: %.c
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(OS_LCFG_CFLAGS)" $< >> Makefile.depend
+
+#
+#  依存関係ファイルの生成
+#
+.PHONY: gendepend
+gendepend:
+	@echo "Generating Makefile.depend."
+
+.PHONY: depend
+ifdef KERNEL_LIB
+depend: cleandep Os_Lcfg.timestamp gendepend \
+		cfg1_out.d \
+		$(ALL_OBJS:.o=.d)
+else
+depend: cleandep $(OFFSET_H) Os_Lcfg.timestamp gendepend \
+		cfg1_out.d \
+		$(KERNEL_AUX_COBJS:.o=.d) $(KERNEL_ASMOBJS:.o=.d) \
+		$(KERNEL_COBJS:.o=.d) $(KERNEL_LCSRCS:.c=.d) $(ALL_OBJS:.o=.d)
+endif
+
+#
+#  依存関係ファイルをインクルード
+#
+-include Makefile.depend
+
+#
+#  開発ツールのコマンド名の定義
+#
+ifeq ($(TOOL),gcc)
+  #
+  #  GNU開発環境用
+  #
+  ifdef GCC_TARGET
+    GCC_TARGET_PREFIX = $(GCC_TARGET)-
+  else
+    GCC_TARGET_PREFIX =
+  endif
+  CC = $(GCC_TARGET_PREFIX)gcc
+  CXX = $(GCC_TARGET_PREFIX)g++
+  AS = $(GCC_TARGET_PREFIX)as
+  LD = $(GCC_TARGET_PREFIX)ld
+  AR = $(GCC_TARGET_PREFIX)ar
+  NM = $(GCC_TARGET_PREFIX)nm
+  RANLIB = $(GCC_TARGET_PREFIX)ranlib
+  OBJCOPY = $(GCC_TARGET_PREFIX)objcopy
+  OBJDUMP = $(GCC_TARGET_PREFIX)objdump
+endif
+
+ifdef USE_CXX
+  LINK = $(CXX)
+else
+  LINK = $(CC)
+endif
+
+#
+#  コンパイルルールの定義
+#
+KERNEL_ALL_COBJS = $(KERNEL_COBJS) $(KERNEL_AUX_COBJS)
+
+$(KERNEL_ALL_COBJS): %.o: %.c
+	$(CC) -c $(CFLAGS) $(KERNEL_CFLAGS) $<
+
+$(KERNEL_ALL_COBJS:.o=.s): %.s: %.c
+	$(CC) -S $(CFLAGS) $(KERNEL_CFLAGS) $<
+
+$(KERNEL_LCOBJS): %.o:
+	$(CC) -DTOPPERS_$(*F) -o $@ -c $(CFLAGS) $(KERNEL_CFLAGS) $<
+
+$(KERNEL_LCOBJS:.o=.s): %.s:
+	$(CC) -DTOPPERS_$(*F) -o $@ -S $(CFLAGS) $(KERNEL_CFLAGS) $<
+
+$(KERNEL_ASMOBJS): %.o: %.S
+	$(CC) -c $(CFLAGS) $(KERNEL_CFLAGS) $<
+
+$(SYSMOD_COBJS): %.o: %.c
+	$(CC) -c $(CFLAGS) $(SYSMOD_CFLAGS) $<
+
+$(SYSMOD_COBJS:.o=.s): %.s: %.c
+	$(CC) -S $(CFLAGS) $(SYSMOD_CFLAGS) $<
+
+$(SYSMOD_ASMOBJS): %.o: %.S
+	$(CC) -c $(CFLAGS) $(SYSMOD_CFLAGS) $<
+
+$(APPL_COBJS): %.o: %.c
+	$(CC) -c $(CFLAGS) $(APPL_CFLAGS) $<
+
+$(APPL_COBJS:.o=.s): %.s: %.c
+	$(CC) -S $(CFLAGS) $(APPL_CFLAGS) $<
+
+$(APPL_CXXOBJS): %.o: %.cpp
+	$(CXX) -c $(CFLAGS) $(APPL_CFLAGS) $<
+
+$(APPL_CXXOBJS:.o=.s): %.s: %.cpp
+	$(CXX) -S $(CFLAGS) $(APPL_CFLAGS) $<
+
+$(APPL_ASMOBJS): %.o: %.S
+	$(CC) -c $(CFLAGS) $(APPL_CFLAGS) $<
+
+#
+#  依存関係作成ルールの定義
+#
+$(KERNEL_COBJS:.o=.d): %.d: %.c
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(KERNEL_CFLAGS)" $< >> Makefile.depend
+
+$(KERNEL_LCSRCS:.c=.d): %.d: %.c
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) -T "$($*)" \
+		-O "-DALLFUNC $(CFLAGS) $(KERNEL_CFLAGS)" $< >> Makefile.depend
+
+$(KERNEL_ASMOBJS:.o=.d): %.d: %.S
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(KERNEL_CFLAGS)" $< >> Makefile.depend
+
+$(SYSMOD_COBJS:.o=.d): %.d: %.c
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(SYSMOD_CFLAGS)" $< >> Makefile.depend
+
+$(SYSMOD_ASMOBJS:.o=.d): %.d: %.S
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(SYSMOD_CFLAGS)" $< >> Makefile.depend
+
+$(APPL_COBJS:.o=.d): %.d: %.c
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(APPL_CFLAGS)" $< >> Makefile.depend
+
+$(APPL_CXXOBJS:.o=.d): %.d: %.cpp
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CXX) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(APPL_CFLAGS)" $< >> Makefile.depend
+
+$(APPL_ASMOBJS:.o=.d): %.d: %.S
+	@$(PERL) $(SRCDIR)/utils/makedep -C $(CC) $(MAKEDEP_OPTS) \
+		-O "$(CFLAGS) $(APPL_CFLAGS)" $< >> Makefile.depend
+
+#
+#  デフォルトコンパイルルールを上書き
+#
+%.o: %.c
+	@echo "*** Default compile rules should not be used."
+	$(CC) -c $(CFLAGS) $<
+
+%.s: %.c
+	@echo "*** Default compile rules should not be used."
+	$(CC) -S $(CFLAGS) $<
+
+%.o: %.cpp
+	@echo "*** Default compile rules should not be used."
+	$(CXX) -c $(CFLAGS) $<
+
+%.s: %.cpp
+	@echo "*** Default compile rules should not be used."
+	$(CXX) -S $(CFLAGS) $<
+
+%.o: %.S
+	@echo "*** Default compile rules should not be used."
+	$(CC) -c $(CFLAGS) $<
+
+
+#
+#  Canジェネレータ
+#
+CAN_PATH = ../../../a-comstack/can
+CAN_ARCH_PATH = $(CAN_PATH)/arch/rs_can
+CAN_TARGET_PATH = $(CAN_PATH)/target/hsbrh850f1k_gcc
+CAN_CFG_TABS = --kernel atk2 --ini-file $(CAN_PATH)/can.ini --api-table $(CAN_PATH)/can.csv
+CAN_CFG_OUT = Can_Cfg.h Can_PBcfg.c
+
+.PHONY: can
+can: $(CAN_CFG_OUT)
+$(CAN_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(CAN_PATH)/can.tf $(CAN_PATH)/can.ini $(CAN_PATH)/can.csv
+	$(CFG) --omit-symbol --pass 2 $(CAN_CFG_TABS) -I $(CAN_PATH) -I $(CAN_ARCH_PATH) -T $(CAN_TARGET_PATH)/Can_Target.tf $(filter %.arxml,$^)
+
+.PHONY: canclean
+canclean:
+	rm -f $(CAN_CFG_OUT)
+
+#
+#  CanIfジェネレータ
+#
+CANIF_PATH = ../../../a-comstack/canif
+CANIF_CFG_TABS = --kernel atk2 --ini-file $(CANIF_PATH)/canif.ini --api-table $(CANIF_PATH)/canif.csv
+CANIF_CFG_OUT = CanIf_Cfg.h CanIf_Lcfg.h CanIf_PBcfg.c CanIf_Lcfg.c
+
+.PHONY: canif
+canif: $(CANIF_CFG_OUT)
+$(CANIF_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(CANIF_PATH)/canif.tf $(CANIF_PATH)/canif.ini $(CANIF_PATH)/canif.csv
+	$(CFG) --omit-symbol --pass 2 $(CANIF_CFG_TABS) -I $(CANIF_PATH) -T $(CANIF_PATH)/canif.tf $(filter %.arxml,$^)
+
+.PHONY: canifclean
+canifclean:
+	rm -f $(CANIF_CFG_OUT)
+
+#
+#  PduRジェネレータ
+#
+PDUR_PATH = ../../../a-comstack/pdur
+PDUR_CFG_TABS = --kernel atk2 --ini-file $(PDUR_PATH)/pdur.ini --api-table $(PDUR_PATH)/pdur.csv
+PDUR_CFG_OUT = PduR_Cfg.h PduR_CanIf.h PduR_Dcm.h PduR_CanTp.h PduR_Com.h
+
+.PHONY: pdur
+pdur: $(PDUR_CFG_OUT)
+$(PDUR_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(PDUR_PATH)/pdur.tf $(PDUR_PATH)/pdur.ini $(PDUR_PATH)/pdur.csv
+	$(CFG) --omit-symbol --pass 2 $(PDUR_CFG_TABS) -I $(PDUR_PATH) -T $(PDUR_PATH)/pdur.tf $(filter %.arxml,$^)
+
+.PHONY: pdurclean
+pdurclean:
+	rm -f $(PDUR_CFG_OUT)
+
+#
+#  Comジェネレータ
+#
+COM_PATH = ../../../a-comstack/com
+COM_CFG_TABS = --kernel atk2 --ini-file $(COM_PATH)/com.ini --api-table $(COM_PATH)/com.csv
+COM_CFG_OUT = Com_Cfg.h Com_PBcfg.c Com_Cbk.h 
+
+.PHONY: com
+com: $(COM_CFG_OUT)
+$(COM_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(COM_PATH)/com.tf $(COM_PATH)/com.ini $(COM_PATH)/com.csv
+	$(CFG) --omit-symbol --pass 2 $(COM_CFG_TABS) -I $(COM_PATH) -T $(COM_PATH)/com.tf $(filter %.arxml,$^)
+
+.PHONY: comclean
+comclean:
+	rm -f $(COM_CFG_OUT)
+
+
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_x/abrex.sh ecu_education_work_step3/swc-application/1ecu/ecu_x/abrex.sh
--- ecu_education_work_step2/swc-application/1ecu/ecu_x/abrex.sh	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_x/abrex.sh	2023-09-11 05:46:35.083684500 +0900
@@ -0,0 +1,4 @@
+OS_DIR=../../../atk2-sc1
+ABREX_DIR=$OS_DIR/utils/abrex
+APP_NAME=Ecuc_X
+ruby $ABREX_DIR/abrex.rb ./$APP_NAME.yaml -e /System/RcCar
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_x/device_config.txt ecu_education_work_step3/swc-application/1ecu/ecu_x/device_config.txt
--- ecu_education_work_step2/swc-application/1ecu/ecu_x/device_config.txt	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_x/device_config.txt	2023-09-11 05:46:35.088739300 +0900
@@ -0,0 +1,6 @@
+DEVICE_CONFIG_TIMER_FD		1
+DEBUG_FUNC_ENABLE_FT		1
+DEBUG_FUNC_ENABLE_BT		1
+DEBUG_FUNC_ENABLE_PROF		1
+DEBUG_FUNC_ENABLE_WATCH		1
+DEBUG_FUNC_ENABLE_SKIP_CLOCK    0
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_x/device_config_with_can.txt ecu_education_work_step3/swc-application/1ecu/ecu_x/device_config_with_can.txt
--- ecu_education_work_step2/swc-application/1ecu/ecu_x/device_config_with_can.txt	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_x/device_config_with_can.txt	2023-09-11 05:46:35.093821200 +0900
@@ -0,0 +1,15 @@
+DEVICE_CONFIG_TIMER_FD		1
+DEBUG_FUNC_ENABLE_FT		1
+DEBUG_FUNC_ENABLE_BT		1
+DEBUG_FUNC_ENABLE_PROF		1
+DEBUG_FUNC_ENABLE_WATCH		1
+DEBUG_FUNC_ENABLE_SKIP_CLOCK    0
+DEBUG_FUNC_ENABLE_CAN      1
+DEVICE_CONFIG_CAN_FD		100000
+DEBUG_FUNC_CAN_RX_QUEUE_SIZE	256
+DEBUG_FUNC_MROS_NODE_NAME	athrill_ecu_x
+DEBUG_FUNC_MROS_SLAVE_PORT_NO	11611
+DEBUG_FUNC_MROS_PUBLISHER_PORT_NO	11711
+DEBUG_FUNC_MROS_TOPIC_PUB_NUM   1
+DEBUG_FUNC_MROS_TOPIC_PUB_0 channel0/CAN_IDE0_RTR0_DLC8_0x001
+
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_x/generator.sh ecu_education_work_step3/swc-application/1ecu/ecu_x/generator.sh
--- ecu_education_work_step2/swc-application/1ecu/ecu_x/generator.sh	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_x/generator.sh	2023-09-11 05:46:35.097818600 +0900
@@ -0,0 +1,6 @@
+OS_DIR=../../../atk2-sc1
+TARGET_DIR=$OS_DIR/target/hsbrh850f1k_gcc
+RTE_DIR=../../../a-rtegen
+ARXML_DIR=../../common/arxmls
+APP_NAME=Ecuc_X
+$RTE_DIR/bin/bin/rtegen.sh $TARGET_DIR/target_hw_counter.arxml $ARXML_DIR/DataTypes.arxml $ARXML_DIR/BswMD_TxRx.arxml ../System_EcuInstance_X.arxml $APP_NAME.arxml
diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_x/memory.txt ecu_education_work_step3/swc-application/1ecu/ecu_x/memory.txt
--- ecu_education_work_step2/swc-application/1ecu/ecu_x/memory.txt	1970-01-01 09:00:00.000000000 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_x/memory.txt	2023-09-11 05:46:35.101820300 +0900
@@ -0,0 +1,7 @@
+ROM, 0x00000000, 1024
+RAM, 0xFEDE8000, 512
+MMAP, 0xFFC10000, mmap1.bin
+MMAP, 0xFFCE5000, mmap2.bin
+MMAP, 0xFFCF0000, mmap3.bin
+MMAP, 0xFFE30000, mmap4.bin
+MMAP, 0xFF305288, mmap5.bin
\ No newline at end of file
diff -uNr ecu_education_work_step2/swc-application/EcuM/EcuM_Callout.h ecu_education_work_step3/swc-application/EcuM/EcuM_Callout.h
--- ecu_education_work_step2/swc-application/EcuM/EcuM_Callout.h	2023-07-02 23:20:53.654070300 +0900
+++ ecu_education_work_step3/swc-application/EcuM/EcuM_Callout.h	2023-09-11 05:46:35.567281200 +0900
@@ -117,5 +117,12 @@
 	 */
 
+#ifdef ECU_X_ALARM_100ms
+	ercd += SetRelAlarm(ECU_X_ALARM_100ms, ALARM_TICK_100MS, ALARM_TICK_100MS);
+#endif /* ECU_X_ALARM_100ms */
+#ifdef ECU_X_ALARM_20ms_SCHM
+	ercd += SetRelAlarm(ECU_X_ALARM_20ms_SCHM, ALARM_TICK_20MS, ALARM_TICK_20MS);
+#endif /* ECU_X_ALARM_20ms_SCHM */
+
 #ifdef ECU_CBO_ALARM_125us
 	ercd += SetRelAlarm(ECU_CBO_ALARM_125us, ALARM_TICK_125US, ALARM_TICK_125US);
 #endif /* ECU_CBO_ALARM_125us */
@@ -125,6 +132,9 @@
 #ifdef ECU_CBO_ALARM_100ms
 	ercd += SetRelAlarm(ECU_CBO_ALARM_100ms, ALARM_TICK_100MS, ALARM_TICK_100MS);
 #endif /* ECU_CBO_ALARM_100ms */
+#ifdef ECU_CBO_ALARM_20ms_SCHM
+	ercd += SetRelAlarm(ECU_CBO_ALARM_20ms_SCHM, ALARM_TICK_20MS, ALARM_TICK_20MS);
+#endif /* ECU_CBO_ALARM_20ms_SCHM */
 
 #ifdef ECU_CB_ALARM_125us
 	ercd += SetRelAlarm(ECU_CB_ALARM_125us, ALARM_TICK_125US, ALARM_TICK_125US);
