diff -uNr ecu_education_work_step2/swc-application/1ecu/ecu_cbo/Makefile ecu_education_work_step3/swc-application/1ecu/ecu_cbo/Makefile
--- ecu_education_work_step2/swc-application/1ecu/ecu_cbo/Makefile	2023-08-30 15:38:44.407388700 +0900
+++ ecu_education_work_step3/swc-application/1ecu/ecu_cbo/Makefile	2023-09-11 05:46:34.993476000 +0900
@@ -191,9 +191,9 @@
 ifndef OMIT_OPTIMIZATION
   COPTS := $(COPTS) -O2
 endif
-USER_MACRO = -DTOPPERS_NO_USE_SCHM -DTOPPERS_NO_USE_COMSTACK -DTOPPERS_USE_BODY -DTOPPERS_USE_OPERATION -DTOPPERS_USE_CONTROL -DSIMULATE_ON_ATHRILL -DECU_EDUCATION_USE_ECU_X
+USER_MACRO = -DTOPPERS_USE_BODY -DTOPPERS_USE_OPERATION -DTOPPERS_USE_CONTROL -DSIMULATE_ON_ATHRILL
 CDEFS := $(CDEFS)  $(USER_MACRO)
-INCLUDES := -I. -I$(SRCDIR)/include -I$(SRCDIR)/arch -I$(SRCDIR) $(INCLUDES)
+INCLUDES := -I. -I$(SRCDIR)/include -I$(SRCDIR)/arch -I$(SRCDIR) $(INCLUDES) -I../../../a-comstack/general
 LDFLAGS := $(LDFLAGS) 
 LIBS := $(LIBS) $(CXXLIBS)
 CFLAGS = $(COPTS) $(CDEFS) $(INCLUDES)
@@ -218,9 +217,24 @@
 	driver_if \
 	rcb3 \
 	rlin3x \
-	taub_pwm 
-
-APPLDIR = ../../../driver ../../common/swcs ../../EcuM
+	taub_pwm \
+	Com \
+	CanIf \
+	Can \
+	Can_Irq \
+	Can_Prc \
+	Com_PBcfg \
+	CanIf_PBcfg \
+	CanIf_Lcfg \
+	Can_PBcfg
+APPLDIR = ../../../driver \
+          ../../common/swcs \
+          ../../EcuM \
+          ../../../a-comstack/com \
+          ../../../a-comstack/canif \
+          ../../../a-comstack/can \
+          ../../../a-comstack/can/target/hsbrh850f1l_ccrh \
+          ../../../a-comstack/can/arch/rs_can
 APPL_CFG_INPUT := $(foreach file,$(CFGNAME),$(file).arxml)
 
 APPL_DIR = $(APPLDIR) $(SRCDIR)/library
@@ -231,7 +245,7 @@
 else
   APPL_COBJS = $(APPLNAME:=.o) 
 endif
-APPL_CFLAGS =
+APPL_CFLAGS =-I../../../a-comstack/general -I../../../a-comstack/can/arch/gcc -I../../../a-comstack/stub
 APPL_LIBS =
 ifdef APPLDIR
   INCLUDES := $(INCLUDES) $(foreach dir,$(APPLDIR),-I$(dir))
@@ -625,3 +639,71 @@
 %.o: %.S
 	@echo "*** Default compile rules should not be used."
 	$(CC) -c $(CFLAGS) $<
+
+#
+#  Canジェネレータ
+#
+CAN_PATH = ../../../a-comstack/can
+CAN_ARCH_PATH = $(CAN_PATH)/arch/rs_can
+CAN_TARGET_PATH = $(CAN_PATH)/target/hsbrh850f1k_gcc
+CAN_CFG_TABS = --kernel atk2 --ini-file $(CAN_PATH)/can.ini --api-table $(CAN_PATH)/can.csv
+CAN_CFG_OUT = Can_Cfg.h Can_PBcfg.c
+
+.PHONY: can
+can: $(CAN_CFG_OUT)
+$(CAN_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(CAN_PATH)/can.tf $(CAN_PATH)/can.ini $(CAN_PATH)/can.csv
+	$(CFG) --omit-symbol --pass 2 $(CAN_CFG_TABS) -I $(CAN_PATH) -I $(CAN_ARCH_PATH) -T $(CAN_TARGET_PATH)/Can_Target.tf $(filter %.arxml,$^)
+
+.PHONY: canclean
+canclean:
+	rm -f $(CAN_CFG_OUT)
+
+#
+#  CanIfジェネレータ
+#
+CANIF_PATH = ../../../a-comstack/canif
+CANIF_CFG_TABS = --kernel atk2 --ini-file $(CANIF_PATH)/canif.ini --api-table $(CANIF_PATH)/canif.csv
+CANIF_CFG_OUT = CanIf_Cfg.h CanIf_Lcfg.h CanIf_PBcfg.c CanIf_Lcfg.c
+
+.PHONY: canif
+canif: $(CANIF_CFG_OUT)
+$(CANIF_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(CANIF_PATH)/canif.tf $(CANIF_PATH)/canif.ini $(CANIF_PATH)/canif.csv
+	$(CFG) --omit-symbol --pass 2 $(CANIF_CFG_TABS) -I $(CANIF_PATH) -T $(CANIF_PATH)/canif.tf $(filter %.arxml,$^)
+
+.PHONY: canifclean
+canifclean:
+	rm -f $(CANIF_CFG_OUT)
+
+#
+#  PduRジェネレータ
+#
+PDUR_PATH = ../../../a-comstack/pdur
+PDUR_CFG_TABS = --kernel atk2 --ini-file $(PDUR_PATH)/pdur.ini --api-table $(PDUR_PATH)/pdur.csv
+PDUR_CFG_OUT = PduR_Cfg.h PduR_CanIf.h PduR_Dcm.h PduR_CanTp.h PduR_Com.h
+
+.PHONY: pdur
+pdur: $(PDUR_CFG_OUT)
+$(PDUR_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(PDUR_PATH)/pdur.tf $(PDUR_PATH)/pdur.ini $(PDUR_PATH)/pdur.csv
+	$(CFG) --omit-symbol --pass 2 $(PDUR_CFG_TABS) -I $(PDUR_PATH) -T $(PDUR_PATH)/pdur.tf $(filter %.arxml,$^)
+
+.PHONY: pdurclean
+pdurclean:
+	rm -f $(PDUR_CFG_OUT)
+
+#
+#  Comジェネレータ
+#
+COM_PATH = ../../../a-comstack/com
+COM_CFG_TABS = --kernel atk2 --ini-file $(COM_PATH)/com.ini --api-table $(COM_PATH)/com.csv
+COM_CFG_OUT = Com_Cfg.h Com_PBcfg.c Com_Cbk.h 
+
+.PHONY: com
+com: $(COM_CFG_OUT)
+$(COM_CFG_OUT): $(APPL_CFG_INPUT) $(TARGETDIR)/target_hw_counter.arxml $(COM_PATH)/com.tf $(COM_PATH)/com.ini $(COM_PATH)/com.csv
+	$(CFG) --omit-symbol --pass 2 $(COM_CFG_TABS) -I $(COM_PATH) -T $(COM_PATH)/com.tf $(filter %.arxml,$^)
+
+.PHONY: comclean
+comclean:
+	rm -f $(COM_CFG_OUT)
+
+
